<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Joyce Hsiao" />


<title>Data overview</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fucci-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/fucci-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Data overview</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>

</div>


<p><strong>Last updated:</strong> 2019-02-12</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(12345)</code> </summary></p>
<p>The command <code>set.seed(12345)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<details>
<p><summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/jdblischak/fucci-seq/tree/391ed92fc2c4058dbf3d8dcd4b7dfb6835a0632d" target="_blank">391ed92</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/method-npreg.Rmd/slurm-46123907.out
    Ignored:    code/method-npreg.Rmd/slurm-46123908.out
    Ignored:    code/method-npreg.Rmd/slurm-46123909.out
    Ignored:    code/method-npreg.Rmd/slurm-46123910.out
    Ignored:    code/method-npreg.Rmd/slurm-46123911.out
    Ignored:    code/method-npreg.Rmd/slurm-46123912.out
    Ignored:    code/method-npreg.Rmd/slurm-46123913.out
    Ignored:    code/method-npreg.Rmd/slurm-46123914.out
    Ignored:    code/method-npreg.Rmd/slurm-46123915.out
    Ignored:    code/method-npreg.Rmd/slurm-46123916.out
    Ignored:    code/npreg-methods.Rmd/slurm-45076734.out
    Ignored:    code/npreg/slurm-45320823.out
    Ignored:    code/trendfilter-individual.Rmd/slurm-49893750.out
    Ignored:    code/trendfilter-individual.Rmd/slurm-49893751.out
    Ignored:    code/trendfilter-individual.Rmd/slurm-49893752.out
    Ignored:    code/trendfilter-individual.Rmd/slurm-49893753.out
    Ignored:    code/trendfilter-individual.Rmd/slurm-49893754.out
    Ignored:    code/trendfilter-individual.Rmd/slurm-49893755.out
    Ignored:    data/batch-paper/
    Ignored:    data/confess-rds/
    Ignored:    data/data-labwebsite/
    Ignored:    data/results/
    Ignored:    dsc/data/
    Ignored:    notes/
    Ignored:    output/npreg-trendfilter.Rmd/
    Ignored:    output_tmp/

Untracked files:
    Untracked:  analysis/cellcycler-seqdata-fucci.Rmd
    Untracked:  analysis/method-eval-bottcher.Rmd
    Untracked:  analysis/method-eval-buettner.Rmd
    Untracked:  analysis/method-eval-leng.Rmd
    Untracked:  analysis/norm-ercc-batch.Rmd
    Untracked:  analysis/norm-ercc-fucci.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</details>
</li>
</ul>
<details>
<summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/fucci-seq/blob/0201c1e495ae8a66b5d8a6258c1a6d26de8a5c48/analysis/data-overview.Rmd" target="_blank">0201c1e</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2019-02-12
</td>
<td style="text-align:left;">
add lab page data link
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/fucci-seq/blob/435d656647e71f27cecd19b95b44d6af635535e6/analysis/data-overview.Rmd" target="_blank">435d656</a>
</td>
<td style="text-align:left;">
Chiaowen Joyce Hsiao
</td>
<td style="text-align:left;">
2019-02-12
</td>
<td style="text-align:left;">
Merge branch ‘master’ into master
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/fucci-seq/435d656647e71f27cecd19b95b44d6af635535e6/docs/data-overview.html" target="_blank">435d656</a>
</td>
<td style="text-align:left;">
Chiaowen Joyce Hsiao
</td>
<td style="text-align:left;">
2019-02-12
</td>
<td style="text-align:left;">
Merge branch ‘master’ into master
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/fucci-seq/blob/9c05421ff4a22a6b1300f44fa91d7ecb5a947a0d/analysis/data-overview.Rmd" target="_blank">9c05421</a>
</td>
<td style="text-align:left;">
Chiaowen Joyce Hsiao
</td>
<td style="text-align:left;">
2019-02-12
</td>
<td style="text-align:left;">
Merge pull request #57 from jdblischak/download
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/fucci-seq/85b83c0fea5921ec75845485678b03de1f3664eb/docs/data-overview.html" target="_blank">85b83c0</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2019-01-31
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/fucci-seq/blob/453d90b64cda43d01a0d65fc28974a3c38a8bc8a/analysis/data-overview.Rmd" target="_blank">453d90b</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2019-01-31
</td>
<td style="text-align:left;">
minor edits
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/fucci-seq/blob/2fd7c02d0ed3a0f9d4e1c6efc44126594ef0af33/analysis/data-overview.Rmd" target="_blank">2fd7c02</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2019-01-31
</td>
<td style="text-align:left;">
minor edits
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/fucci-seq/e87325cea293034ce4b86ff3a19c2fec1cc73a84/docs/data-overview.html" target="_blank">e87325c</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2018-04-11
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/fucci-seq/blob/816dccd8df38b9cf3b0412456d02c05a3a21c321/analysis/data-overview.Rmd" target="_blank">816dccd</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2018-04-11
</td>
<td style="text-align:left;">
tidy up data processing overview
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/fucci-seq/19db0c4b2dea37710c6d16bb67f30390716b182d/docs/data-overview.html" target="_blank">19db0c4</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2018-04-10
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/fucci-seq/blob/77d10d453e1b28dbb5f4f50085734cec91f93e71/analysis/data-overview.Rmd" target="_blank">77d10d4</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2018-04-10
</td>
<td style="text-align:left;">
save final dataset to data/eset-final.rds: add estimated cell time to
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/fucci-seq/28cf63f7c5c61d9d8afba655dcfdcc79ce9d8b36/docs/data-overview.html" target="_blank">28cf63f</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2018-04-09
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/fucci-seq/blob/a1584da26d9f8cd799d7dc4f22db73b8207c55d2/analysis/data-overview.Rmd" target="_blank">a1584da</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2018-01-31
</td>
<td style="text-align:left;">
edits
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/fucci-seq/blob/78d5a2cab20f05df9b8cc482f2b594a80bd99332/analysis/data-overview.Rmd" target="_blank">78d5a2c</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2018-01-16
</td>
<td style="text-align:left;">
data description
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/fucci-seq/b5c6d55e17116e41459566cafedee06e6f0645d0/docs/data-overview.html" target="_blank">b5c6d55</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2017-12-23
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
html
</td>
<td style="text-align:left;">
<a href="https://cdn.rawgit.com/jdblischak/fucci-seq/820a38ae0f32c8c7b5a919cd5e52ec72cd9ee156/docs/data-overview.html" target="_blank">820a38a</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2017-12-13
</td>
<td style="text-align:left;">
Build site.
</td>
</tr>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/jdblischak/fucci-seq/blob/7509725c34a8a27ba0f2eca01ccb672e352fa32b/analysis/data-overview.Rmd" target="_blank">7509725</a>
</td>
<td style="text-align:left;">
Joyce Hsiao
</td>
<td style="text-align:left;">
2017-12-13
</td>
<td style="text-align:left;">
wflow_publish(“analysis/data-overview.Rmd”)
</td>
</tr>
</tbody>
</table>
</ul>
</details>
<hr />
<div id="data-availability" class="section level2">
<h2>Data availability</h2>
<ul>
<li><p>This document lists the datasets analyzed in the study.</p></li>
<li><p>We stored all datasets as <code>expressionSets</code> (require <code>Biobase</code> package).</p></li>
<li><p>We provided data in TXT format on the <a href="https://giladlab.uchicago.edu/wp-content/uploads/2019/02/Hsiao_et_al_2019.tar.gz">Gilad lab website</a> for the 11,093 genes analyzed in the study and for each of the 888 samples: molecule count, sample phenotpyes, gene information, phenotype label descriptiond and FUCCI intensity data.</p></li>
</ul>
</div>
<div id="data-structure" class="section level2">
<h2>Data structure</h2>
<p>We collected two types of data for each single cell sample: single-cell RNA-seq using C1 plates and FUCCI image intensity data.</p>
<ul>
<li><p>Raw RNA-seq data: <a href="https://github.com/jdblischak/fucci-seq/blob/master/data/eset-raw.rds?raw=true"><code>data/eset-raw.rds</code></a></p></li>
<li><p>Filtered RNA-seq data: <a href="https://github.com/jdblischak/fucci-seq/blob/master/data/eset-filtered.rds?raw=true"><code>data/eset-filtered.rds</code></a></p></li>
<li><p>FUCCI intensity data: <a href="https://github.com/jdblischak/fucci-seq/blob/master/data/intensity.rds?raw=true"><code>data/intensity.rds</code></a></p></li>
<li><p>FUCCI intensity data adjusted for batch effect: <a href="https://github.com/jdblischak/fucci-seq/blob/master/output/images-normalize-anova.Rmd/pdata.adj.rds?raw=true"><code>output/images-normalize-anova.Rmd/pdata.adj.rds</code></a></p></li>
<li><p>Final data combining filtered intensity and RNA-seq, including 11093 genes and 888 samples: <a href="https://github.com/jdblischak/fucci-seq/blob/master/data/eset-final.rds?raw=true"><code>data/eset-final.rds</code></a></p></li>
</ul>
<p>Code used to generate data from <code>data/eset-raw.rds</code> to <code>data/eset-final.rds</code> is stored in <code>code/output-raw-2-final.R</code>.</p>
</div>
<div id="downloading-the-data-files" class="section level2">
<h2>Downloading the data files</h2>
<p>You have two main options for downloading the data files. First, you can manually download the individual files by clicking on the links on this page or navigating to the files in the <a href="https://github.com/jdblischak/fucci-seq">fucci-seq</a> GitHub repository. This is the recommended strategy if you only need a few data files.</p>
<p>Second, you can install <a href="https://git-lfs.github.com/">git-lfs</a>. To handle large files, we used Git Large File Storage (LFS). This means that the files that you download with <code>git clone</code> are only plain text files that contain identifiers for the files saved on GitHub’s servers. If you want to download all of the data files at once, you can do this with after you install git-lfs.</p>
<p>To install git-lfs, follow their instructions to download, install, and setup (<code>git lfs install</code>). Alternatively, if you use conda, you can install git-lfs with <code>conda install -c conda-forge git-lfs</code>. Once installed, you can download the latest version of the data files with <code>git lfs pull</code>.</p>
</div>
<div id="how-to-access-expressionsets" class="section level2">
<h2>How to access expressionSets</h2>
<p>We store feature-level (gene) read count and molecule count in <code>expressionSet</code> (<code>data/eset</code>) objects, which also contain sample metadata (e.g., assigned indivdual ID, cDNA concentraion) and quality filtering criteria (e.g., number of reads mapped to FUCCI transgenes, ERCC conversion rate). Data from different C1 plates are stored in separate <code>eset</code> objects:</p>
<p>To combine <code>eset</code> objects from the different C1 plates:</p>
<p><code>eset &lt;- Reduce(combine, Map(readRDS, Sys.glob(&quot;data/eset/*.rds&quot;)))</code></p>
<p>To access data stored in <code>expressionSet</code>:</p>
<ul>
<li><p><code>exprs(eset)</code>: access count data, 20,421 features by 1,536 single cell samples.</p></li>
<li><p><code>pData(eset)</code>: access sample metadata. Returns data.frame of 1,536 samples by 43 labels. Use <code>varMetadata(phenoData(eset))</code> to view label descriptions.</p></li>
<li><p><code>fData(eset)</code>: access feature metadata. Returns data.frame of 20,421 features by 6 labels. Use <code>varMetadata(featureData(eset))</code> to view label descriptions.</p></li>
<li><p><code>varMetadata(phenoData(eset))</code>: view the sample metadata labels.</p></li>
<li><p><code>varMetadata(featureData(eset))</code>: view the feature (gene) metadata labels.</p></li>
</ul>
</div>
<div id="additional-data-information" class="section level2">
<h2>Additional data information</h2>
<div id="fucci-intensity-data" class="section level3">
<h3>FUCCI intensity data</h3>
<ul>
<li><p>Combined intensity data are stored in <code>data/intensity.rds</code>. These include samples that were identified to have a single nuclei .</p></li>
<li><p>Data generated by <a href="data/combine-intensity-data.R">combine-intensity-data.R</a>. Combining image analysis output stored in <code>/project2/gilad/fucci-seq/intensities_stats/</code> into one <code>data.frame</code> and computes summary statistics, including background-corrected RFP and GFP intensity measures.</p></li>
</ul>
</div>
<div id="sequencing-data" class="section level3">
<h3>Sequencing data</h3>
<ul>
<li><p>Raw data from each C1 plate are stored separatley in <code>data/eset/</code> by experiment (batch) ID.</p></li>
<li><p>Raw data combining C1 plate are stored in <code>data/eset-raw.rds</code>.</p></li>
<li><p>Filtered raw data excluding low-quality sequencing samples and genes that are lowly expressed or overly expressed are stored in <code>data/eset-filtered.rds</code>.</p></li>
</ul>
</div>
<div id="phenotypic-data-of-singleton-samples" class="section level3">
<h3>Phenotypic data of singleton samples</h3>
<ul>
<li>Data file 1: all 1536 samples before filtering (<a href="https://raw.githubusercontent.com/jdblischak/fucci-seq/master/output/data-overview.Rmd/phenotypes_allsamples.txt"><code>output/data-overview.Rmd/phenotypes_allsamples.txt</code></a>)</li>
<li>Data file 2: 888 samples after filtering (<a href="https://raw.githubusercontent.com/jdblischak/fucci-seq/master/output/data-overview.Rmd/phenotypes_singletonsamples.txt"><code>output/data-overview.Rmd/phenotypes_singletonsamples.txt</code></a>)</li>
<li>Data file 3: phenotype labels (<a href="https://raw.githubusercontent.com/jdblischak/fucci-seq/master/output/data-overview.Rmd/phenotypes_labels.txt"><code>output/data-overview.Rmd/phenotypes_labels.txt</code></a>)</li>
</ul>
<pre class="r"><code>library(Biobase)

eset_raw &lt;- readRDS(&quot;../data/eset-raw.rds&quot;)
df &lt;- data.frame(sample_id=rownames(pData(eset_raw)), pData(eset_raw), stringsAsFactors = F)
write.table(df, quote=F, sep=&quot;\t&quot;, 
            row.names = F, col.names = T,
            file = &quot;../output/data-overview.Rmd/phenotypes_allsamples.txt&quot;)


eset_final &lt;- readRDS(&quot;../data/eset-final.rds&quot;)
df &lt;- data.frame(sample_id=rownames(pData(eset_final)), pData(eset_final), stringsAsFactors = F)
write.table(df, quote=F, sep=&quot;\t&quot;, 
            row.names = F, col.names = T,
            file = &quot;../output/data-overview.Rmd/phenotypes_singletonsamples.txt&quot;)

labels &lt;- data.frame(var_names=rownames(varMetadata(eset_raw)),
                       labels=varMetadata(eset_raw)$labeDescription, stringsAsFactors = F)
labels &lt;- rbind(labels,
                data.frame(var_names=rownames(varMetadata(eset_final)),
                       labels=varMetadata(eset_final)$labelDescription, stringsAsFactors = F)[45:54,])
write.table(labels, quote=F,
            sep=&quot;\t&quot;, row.names = F, col.names = T,
            file = &quot;../output/data-overview.Rmd/phenotypes_labels.txt&quot;)

# testing reading files
library(data.table)
df_all &lt;- fread(file = &quot;../output/data-overview.Rmd/phenotypes_allsamples.txt&quot;)
df_singles &lt;- fread(file = &quot;../output/data-overview.Rmd/phenotypes_singletonsamples.txt&quot;)
df_labels &lt;- fread(file = &quot;../output/data-overview.Rmd/phenotypes_labels.txt&quot;)</code></pre>
<pre class="r"><code>eset_final &lt;- readRDS(&quot;../data/eset-final.rds&quot;)

df &lt;- data.frame(sample_id=rownames(pData(eset_raw)), pData(eset_raw), stringsAsFactors = F)

write.table(df, quote=F, sep=&quot;\t&quot;, quote=F,
            row.names = F, col.names = T,
            file = &quot;../output/data-overview.Rmd/phenotypes_allsamples.txt&quot;)

write.table(data.frame(var_names=rownames(varMetadata(eset_raw)),
                       labels=varMetadata(eset_raw)$labeDescription, stringsAsFactors = F), 
            quote=F,
            sep=&quot;\t&quot;, row.names = F, col.names = T,
            file = &quot;../output/data-overview.Rmd/phenotypes_allsamples_labels.txt&quot;)

# testing reading files
library(data.table)
df_test &lt;- fread(file = &quot;../output/data-overview.Rmd/phenotypes_allsamples.txt&quot;)
df_labels_test &lt;- fread(file = &quot;../output/data-overview.Rmd/phenotypes_allsamples_labels.txt&quot;)</code></pre>
</div>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] workflowr_1.1.1   Rcpp_1.0.0        digest_0.6.18    
 [4] rprojroot_1.3-2   R.methodsS3_1.7.1 backports_1.1.2  
 [7] magrittr_1.5      git2r_0.23.0      evaluate_0.12    
[10] stringi_1.2.4     whisker_0.3-2     R.oo_1.22.0      
[13] R.utils_2.7.0     rmarkdown_1.10    tools_3.5.1      
[16] stringr_1.3.1     yaml_2.2.0        compiler_3.5.1   
[19] htmltools_0.3.6   knitr_1.20       </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->
<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
