<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Joyce Hsiao" />


<title>Data overview</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fucci-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/fucci-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Data overview</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>

</div>

<div id="TOC">
<ul>
<li><a href="#packages">Packages</a></li>
<li><a href="#overview">Overview</a></li>
<li><a href="#fucci-intensity-data">FUCCI intensity data</a></li>
<li><a href="#sequencing-data">Sequencing data</a></li>
<li><a href="#match-intensity-with-sequencing-data">Match intensity with sequencing data</a></li>
<li><a href="#access-expressionsets">Access expressionSets</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-04-10</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 77d10d4</p>
<hr />
<div id="packages" class="section level2">
<h2>Packages</h2>
<pre class="r"><code>library(Biobase)</code></pre>
<hr />
</div>
<div id="overview" class="section level2">
<h2>Overview</h2>
<p>We collected two types of data for each single cell sample: single-cell RNA-seq using C1 plates and FUCCI image intensity data.</p>
<ul>
<li><p>Raw RNA-seq data: <code>data/eset-raw.rds</code></p></li>
<li><p>FUCCI intensity data: <code>data/intensity.rds</code></p></li>
<li><p>Filtered RNA-seq data: <code>output/gene-filtering.Rmd/eset-gene-filtering.rds</code></p></li>
<li><p>Final data combining filtered intensity and RNA-seq, including 12792 genes and 990 samples: <code>data/eset-final.rds</code></p></li>
</ul>
<hr />
</div>
<div id="fucci-intensity-data" class="section level2">
<h2>FUCCI intensity data</h2>
<ul>
<li><p>Combined intensity data are stored in <code>data/intensity.rds</code>. These include samples that were identified to have a single nuclei .</p></li>
<li><p>Data generated by [combine-intensity-data.R][data/combine-intensity-data.R]. Combining image analysis output stored in <code>/project2/gilad/fucci-seq/intensities_stats/</code> into one <code>data.frame</code> and computes summary statistics, including background-corrected RFP and GFP intensity measures.</p></li>
</ul>
<pre class="r"><code>ints &lt;- readRDS(file=&quot;../data/intensity.rds&quot;)
colnames(ints)</code></pre>
<pre><code> [1] &quot;plate&quot;                 &quot;image&quot;                
 [3] &quot;size&quot;                  &quot;perimeter&quot;            
 [5] &quot;eccentricity&quot;          &quot;rfp.fore.zoom.mean&quot;   
 [7] &quot;rfp.fore.zoom.median&quot;  &quot;gfp.fore.zoom.mean&quot;   
 [9] &quot;gfp.fore.zoom.median&quot;  &quot;dapi.fore.zoom.mean&quot;  
[11] &quot;dapi.fore.zoom.median&quot; &quot;rfp.back.zoom.mean&quot;   
[13] &quot;rfp.back.zoom.median&quot;  &quot;gfp.back.zoom.mean&quot;   
[15] &quot;gfp.back.zoom.median&quot;  &quot;dapi.back.zoom.mean&quot;  
[17] &quot;dapi.back.zoom.median&quot; &quot;rfp.mean.log10sum&quot;    
[19] &quot;gfp.mean.log10sum&quot;     &quot;dapi.mean.log10sum&quot;   
[21] &quot;rfp.median.log10sum&quot;   &quot;gfp.median.log10sum&quot;  
[23] &quot;dapi.median.log10sum&quot;  &quot;unique&quot;               
[25] &quot;chip_id&quot;              </code></pre>
<hr />
</div>
<div id="sequencing-data" class="section level2">
<h2>Sequencing data</h2>
<ul>
<li><p>Raw data from each C1 plate are stored separatley in <code>data/eset/</code> by experiment (batch) ID.</p></li>
<li><p>Raw data combining C1 plate are stored in <code>data/eset-raw.rds</code>.</p></li>
<li><p>Filtered raw data excluding low-quality sequencing samples and genes that are lowly expressed or overly expressed are stored in <code>output/gene-filtering.Rmd/eset-gene-filtering.rds</code> (both eSet and CPM data.table).</p></li>
</ul>
<pre class="r"><code>eset_filtered &lt;- readRDS(&quot;../output/gene-filtering.Rmd/eset-gene-filtering.rds&quot;)</code></pre>
<hr />
</div>
<div id="match-intensity-with-sequencing-data" class="section level2">
<h2>Match intensity with sequencing data</h2>
<p>We match labels in the intensity data and in the sequencing data. 990 samples are quantified in both datasets.</p>
<pre class="r"><code>pdata &lt;- pData(eset_filtered)
pdata$unique &lt;- paste(pdata$image_individual, sprintf(&quot;%05d&quot;, pdata$image_label), sep=&quot;_&quot;)

sample_include_bothdata &lt;- intersect(ints$unique, pdata$unique)
length(sample_include_bothdata)</code></pre>
<pre><code>[1] 990</code></pre>
<p>Make a combined eSet object include both FUCCI intensity data and RNA-seq data.</p>
<pre class="r"><code>ints_combo &lt;- ints[which(ints$unique %in% sample_include_bothdata), ]

eset_combo &lt;- new(&quot;ExpressionSet&quot;, 
      exprs = exprs(eset_filtered)[,which(pdata$unique %in% sample_include_bothdata)], 
      phenoData = phenoData(eset_filtered)[which(pdata$unique %in% sample_include_bothdata), ], 
      featureData = featureData(eset_filtered))

pdata_combo &lt;- pData(eset_combo)
pdata_combo$unique &lt;- paste(pdata_combo$image_individual, 
                            sprintf(&quot;%05d&quot;, pdata_combo$image_label), sep=&quot;_&quot;)

all.equal(ints_combo$unique, pdata_combo$unique)</code></pre>
<p>Import estimated cell time. Computed using cell times adjusted for plate effect <code>../output/images-normalize-anova.Rmd/pdata.adj.rds</code>. See <code>../docs/npreg.html</code> for methods.</p>
<pre class="r"><code>theta &lt;- readRDS(file = &quot;../output/npreg.Rmd/theta.rds&quot;)</code></pre>
<p>Make variable labels.</p>
<pre class="r"><code>pdata_table &lt;- rbind(varMetadata(phenoData(eset_combo)),
                    &quot;mCherry background-corrected intensity (log10sum)&quot;,
                    &quot;EGFP background-corrected intensity (log10sum)&quot;,
                    &quot;DAPI background-corrected intensity (log10sum)&quot;,
                    &quot;nucleus size&quot;,
                    &quot;nucleus perimeter&quot;,
                    &quot;nucleus eccentricity&quot;, 
                    &quot;cell time&quot;)
rownames(pdata_table) &lt;- c(rownames(varMetadata(phenoData(eset_combo))),
                           &quot;rfp.median.log10sum&quot;,
                           &quot;gfp.median.log10sum&quot;,
                           &quot;dapi.median.log10sum&quot;,
                           &quot;size&quot;, &quot;perimeter&quot;, &quot;eccentricity&quot;,
                           &quot;theta&quot;)


phenoData(eset_combo) &lt;- new(&quot;AnnotatedDataFrame&quot;, 
  data = data.frame(pData(eset_combo),
  rfp.median.log10sum=ints_combo$rfp.median.log10sum,
  gfp.median.log10sum=ints_combo$gfp.median.log10sum,                               dapi.median.log10sum=ints_combo$dapi.median.log10sum,
  size=ints_combo$size,
  perimeter=ints_combo$perimeter,
  eccentricity=ints_combo$eccentricity,
  theta=theta),
  varMetadata = pdata_table)

saveRDS(eset_combo, file = &quot;../data/eset-final.rds&quot;)</code></pre>
<hr />
</div>
<div id="access-expressionsets" class="section level2">
<h2>Access expressionSets</h2>
<p>We store feature-level (gene) read count and molecule count in <code>expressionSet</code> (<code>data/eset</code>) objects, which also contain sample metadata (e.g., assigned indivdual ID, cDNA concentraion) and quality filtering criteria (e.g., number of reads mapped to FUCCI transgenes, ERCC conversion rate). Data from different C1 plates are stored in separate <code>eset</code> objects:</p>
<p>To combine <code>eset</code> objects from the different C1 plates:</p>
<p><code>eset &lt;- Reduce(combine, Map(readRDS, Sys.glob(&quot;data/eset/*.rds&quot;)))</code></p>
<p>To access data stored in <code>expressionSet</code>:</p>
<ul>
<li><p><code>exprs(eset)</code>: access count data, 20,421 features by 1,536 single cell samples.</p></li>
<li><p><code>pData(eset)</code>: access sample metadata. Returns data.frame of 1,536 samples by 43 labels. Use <code>varMetadata(phenoData(eset))</code> to view label descriptions.</p></li>
<li><p><code>fData(eset)</code>: access feature metadata. Returns data.frame of 20,421 features by 6 labels. Use <code>varMetadata(featureData(eset))</code> to view label descriptions.</p></li>
<li><p><code>varMetadata(phenoData(eset))</code>: view the sample metadata labels.</p></li>
<li><p><code>varMetadata(featureData(eset))</code>: view the feature (gene) metadata labels.</p></li>
</ul>
<hr />
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre><code>R version 3.4.1 (2017-06-30)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /usr/lib64/R/lib/libRblas.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] Biobase_2.38.0      BiocGenerics_0.24.0

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.16    digest_0.6.15   rprojroot_1.3-2 backports_1.1.2
 [5] git2r_0.21.0    magrittr_1.5    evaluate_0.10.1 stringi_1.1.7  
 [9] rmarkdown_1.9   tools_3.4.1     stringr_1.3.0   yaml_2.1.18    
[13] compiler_3.4.1  htmltools_0.3.6 knitr_1.20     </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
