<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Joyce Hsiao" />


<title>Correlation between intensities and expression</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fucci-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/fucci-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Correlation between intensities and expression</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-02-23</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 4b23dde</p>
<hr />
<div id="overviewresults" class="section level2">
<h2>Overview/Results</h2>
<p>I computed correlation between intensity measurements and gene expression (log2cpm). Then apply CorShrink to account for varying sample size between correlations.</p>
<p>Results show that there are some indication of cell cycle signal in teh data. Of the genes with s-vaule &lt; .01, 33 correlate with DAPI are annotated as cell cycle gene, 74 correlation with RFP and annotated as cell cycle gene, and 110 correlate with GFP are annotated as cell cycle gene. However, these numbers are not large consider that there are a total of 11429 genes in the data, and among these only 1055 are annotated as cell cycle gene.</p>
<p>In addition, there are sources of variation that may drive sample differences other than cell cycle, and crucially our design induces a strong sample variation between batches, as a trade-off for avoiding confounding of individual and plate effect. Hence analysis of sample variation in our data should consider technical sources of variation along with biological variation.</p>
<hr />
</div>
<div id="data-and-packages" class="section level2">
<h2>Data and packages</h2>
<p>Packages</p>
<pre class="r"><code>library(CorShrink)
library(mygene)</code></pre>
<p>Load data</p>
<pre class="r"><code>df &lt;- readRDS(file=&quot;../data/eset-filtered.rds&quot;)
pdata &lt;- pData(df)
fdata &lt;- fData(df)

# select endogeneous genes
counts &lt;- exprs(df)[grep(&quot;ENSG&quot;, rownames(df)), ]

# import corrected intensities
pdata.adj &lt;- readRDS(&quot;../output/images-normalize-anova.Rmd/pdata.adj.rds&quot;)

log2cpm &lt;- readRDS(&quot;../output/seqdata-batch-correction.Rmd/log2cpm.rds&quot;)
log2cpm.adjust &lt;- readRDS(&quot;../output/seqdata-batch-correction.Rmd/log2cpm.adjust.rds&quot;)

log2cpm &lt;- log2cpm[grep(&quot;ENSG&quot;, rownames(log2cpm)), 
                   colnames(log2cpm) %in% rownames(pdata.adj)]

log2cpm.adjust &lt;- log2cpm.adjust[grep(&quot;ENSG&quot;, rownames(log2cpm)),
                   colnames(log2cpm.adjust) %in% rownames(pdata.adj)]

all.equal(rownames(pdata.adj), colnames(log2cpm))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>genes.variable &lt;- readRDS(&quot;../output/seqdata-select-cellcyclegenes.Rmd/genes.variable.rds&quot;)
genes.cycle &lt;- readRDS(file = &quot;../output/seqdata-select-cellcyclegenes.Rmd/genes.cycle.rds&quot;)</code></pre>
</div>
<div id="correlations" class="section level2">
<h2>Correlations</h2>
<p>compute correlation between adjusted intensities and log2 expression data.</p>
<pre class="r"><code>corr.rfp &lt;- do.call(rbind, lapply(1:nrow(log2cpm), function(i) {
  vec &lt;- cbind(pdata.adj$rfp.median.log10sum.adjust.ash,
               log2cpm[i,])
  filt &lt;- counts[i,] &gt; 1
  nsamp &lt;- sum(filt)
  if (nsamp &gt; 100) {
#    cnt &lt;- counts[i,filt]
    vec &lt;- vec[filt,]
    corr &lt;- cor(vec[,1], vec[,2])
    nsam &lt;- nrow(vec)
    data.frame(corr=corr, nsam=nsam)
  } else {
    data.frame(corr=NA, nsam=nrow(vec))
  }
  }) )


corr.gfp &lt;- do.call(rbind, lapply(1:nrow(log2cpm), function(i) {
  vec &lt;- cbind(pdata.adj$gfp.median.log10sum.adjust.ash,
               log2cpm[i,])
  filt &lt;- counts[i,] &gt; 1
  nsamp &lt;- sum(filt)
  if (nsamp &gt; 100) {
    vec &lt;- vec[filt,]
    corr &lt;- cor(vec[,1], vec[,2])
    nsam &lt;- nrow(vec)
    data.frame(corr=corr, nsam=nsam)
  } else {
    data.frame(corr=NA, nsam=nrow(vec))
  }
  }) )


corr.dapi &lt;- do.call(rbind, lapply(1:nrow(log2cpm), function(i) {
  vec &lt;- cbind(pdata.adj$dapi.median.log10sum.adjust.ash,
               log2cpm[i,])
  filt &lt;- counts[i,] &gt; 1
  nsamp &lt;- sum(filt)
  if (nsamp &gt; 100) {
    vec &lt;- vec[filt,]
    corr &lt;- cor(vec[,1], vec[,2])
    nsam &lt;- nrow(vec)
    data.frame(corr=corr, nsam=nsam)
  } else {
    data.frame(corr=NA, nsam=nrow(vec))
  }
  }) )
rownames(corr.rfp) &lt;- rownames(log2cpm)
rownames(corr.gfp) &lt;- rownames(log2cpm)
rownames(corr.dapi) &lt;- rownames(log2cpm)


corr.rfp.val &lt;- corr.rfp[!is.na(corr.rfp$corr),]
corr.gfp.val &lt;- corr.gfp[!is.na(corr.gfp$corr),]
corr.dapi.val &lt;- corr.dapi[!is.na(corr.dapi$corr),]


par(mfrow=c(2,2))
hist(corr.rfp.val$corr, main = &quot;correlation between RFP and expression&quot;)
hist(corr.gfp.val$corr, main = &quot;correlation between GFP and expression&quot;)
hist(corr.dapi.val$corr, main = &quot;correlation between DAPI and expression&quot;)</code></pre>
<p><img src="figure/images-seq-correlation.Rmd/unnamed-chunk-3-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>Apply CorShrink.</p>
<pre class="r"><code>corr.rfp.shrink &lt;- CorShrinkVector(corr.rfp.val$corr, nsamp_vec = corr.rfp.val$nsam,
                                   optmethod = &quot;mixEM&quot;, report_model = TRUE)
names(corr.rfp.shrink$estimate) &lt;- rownames(corr.rfp.val)

corr.gfp.shrink &lt;- CorShrinkVector(corr.gfp.val$corr, nsamp_vec = corr.gfp.val$nsam,
                                   optmethod = &quot;mixEM&quot;, report_model = TRUE)
names(corr.gfp.shrink$estimate) &lt;- rownames(corr.gfp.val)

corr.dapi.shrink &lt;- CorShrinkVector(corr.dapi.val$corr, nsamp_vec = corr.dapi.val$nsam,
                                   optmethod = &quot;mixEM&quot;, report_model = TRUE)
names(corr.dapi.shrink$estimate) &lt;- rownames(corr.dapi.val)


par(mfcol=c(2,2))
plot(corr.rfp.val$corr, corr.rfp.shrink$estimate,
     col=1+as.numeric(corr.rfp.shrink$model$result$svalue &lt; .01),
     xlim=c(-.5,.5), ylim=c(-.5,.5), pch=16, cex=.8,
     xlab = &quot;Correlation&quot;, ylab = &quot;Shrunken estimate&quot;,
     main = &quot;RFP&quot;)
abline(0,1)

plot(corr.gfp.val$corr, corr.gfp.shrink$estimate,
     col=1+as.numeric(corr.gfp.shrink$model$result$svalue &lt; .01),
     xlim=c(-.5,.5), ylim=c(-.5,.5), pch=16, cex=.8,
     xlab = &quot;Correlation&quot;, ylab = &quot;Shrunken estimate&quot;,
     main = &quot;GFP&quot;)
abline(0,1)

plot(corr.dapi.val$corr, corr.dapi.shrink$estimate,
     col=1+as.numeric(corr.dapi.shrink$model$result$svalue &lt; .01),
     xlim=c(-.5,.5), ylim=c(-.5,.5), pch=16, cex=.8,
     xlab = &quot;Correlation&quot;, ylab = &quot;Shrunken estimate&quot;,
     main = &quot;DAPI&quot;)
abline(0,1)
title(&quot;s-value &lt; .01&quot;, outer = TRUE, line = -1)</code></pre>
<p><img src="figure/images-seq-correlation.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Of the ones with s-value &lt; .05, consider the proportion of genes in our annotated set.</p>
<pre class="r"><code>ii.dapi &lt;- names(corr.dapi.shrink$estimate)[which(corr.dapi.shrink$model$result$svalue &lt; .01)]
print(sum(ii.dapi %in% genes.cycle))</code></pre>
<pre><code>[1] 33</code></pre>
<pre class="r"><code>print(sum(ii.dapi %in% genes.variable))</code></pre>
<pre><code>[1] 20</code></pre>
<pre class="r"><code>ii.rfp &lt;- names(corr.rfp.shrink$estimate)[which(corr.rfp.shrink$model$result$svalue &lt; .01)]
print(sum(ii.rfp %in% genes.cycle))</code></pre>
<pre><code>[1] 74</code></pre>
<pre class="r"><code>print(sum(ii.rfp %in% genes.variable))</code></pre>
<pre><code>[1] 29</code></pre>
<pre class="r"><code>ii.gfp &lt;- names(corr.gfp.shrink$estimate)[which(corr.gfp.shrink$model$result$svalue &lt; .01)]
print(sum(ii.gfp %in% genes.cycle))</code></pre>
<pre><code>[1] 110</code></pre>
<pre class="r"><code>print(sum(ii.gfp %in% genes.variable))</code></pre>
<pre><code>[1] 77</code></pre>
<hr />
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre><code>R version 3.4.1 (2017-06-30)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux 7.2 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /usr/lib64/R/lib/libRblas.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] mygene_1.14.0          GenomicFeatures_1.30.3 AnnotationDbi_1.40.0  
 [4] Biobase_2.38.0         GenomicRanges_1.30.2   GenomeInfoDb_1.14.0   
 [7] IRanges_2.12.0         S4Vectors_0.16.0       BiocGenerics_0.24.0   
[10] CorShrink_0.1.1       

loaded via a namespace (and not attached):
 [1] httr_1.3.1                 RMySQL_0.10.13            
 [3] bit64_0.9-7                jsonlite_1.5              
 [5] splines_3.4.1              foreach_1.4.4             
 [7] gsubfn_0.6-6               Formula_1.2-2             
 [9] assertthat_0.2.0           latticeExtra_0.6-28       
[11] blob_1.1.0                 GenomeInfoDbData_1.0.0    
[13] Rsamtools_1.30.0           yaml_2.1.16               
[15] progress_1.1.2             pillar_1.1.0              
[17] RSQLite_2.0                backports_1.1.2           
[19] lattice_0.20-35            chron_2.3-52              
[21] digest_0.6.15              checkmate_1.8.5           
[23] RColorBrewer_1.1-2         XVector_0.18.0            
[25] colorspace_1.3-2           htmltools_0.3.6           
[27] Matrix_1.2-10              plyr_1.8.4                
[29] XML_3.98-1.10              biomaRt_2.34.2            
[31] zlibbioc_1.24.0            scales_0.5.0              
[33] BiocParallel_1.12.0        htmlTable_1.11.2          
[35] git2r_0.21.0               tibble_1.4.2              
[37] sqldf_0.4-11               ggplot2_2.2.1             
[39] SummarizedExperiment_1.8.1 ashr_2.2-4                
[41] nnet_7.3-12                lazyeval_0.2.1            
[43] proto_1.0.0                survival_2.41-3           
[45] magrittr_1.5               memoise_1.1.0             
[47] evaluate_0.10.1            doParallel_1.0.11         
[49] MASS_7.3-47                foreign_0.8-69            
[51] truncnorm_1.0-7            data.table_1.10.4-3       
[53] tools_3.4.1                prettyunits_1.0.2         
[55] matrixStats_0.53.1         stringr_1.3.0             
[57] munsell_0.4.3              cluster_2.0.6             
[59] DelayedArray_0.4.1         Biostrings_2.46.0         
[61] compiler_3.4.1             rlang_0.2.0               
[63] grid_3.4.1                 RCurl_1.95-4.10           
[65] rstudioapi_0.7             iterators_1.0.9           
[67] htmlwidgets_1.0            bitops_1.0-6              
[69] base64enc_0.1-3            rmarkdown_1.8             
[71] gtable_0.2.0               codetools_0.2-15          
[73] DBI_0.7                    reshape2_1.4.3            
[75] R6_2.2.2                   gridExtra_2.3             
[77] GenomicAlignments_1.14.1   knitr_1.20                
[79] rtracklayer_1.38.3         bit_1.1-12                
[81] Hmisc_4.1-1                rprojroot_1.3-2           
[83] stringi_1.1.6              pscl_1.5.2                
[85] SQUAREM_2017.10-1          Rcpp_0.12.15              
[87] rpart_4.1-11               acepack_1.4.1             </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
