<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Joyce Hsiao" />


<title>Classify cells based on FUCCI: compare PAM vs. Mclust</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fucci-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/fucci-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Classify cells based on FUCCI: compare PAM vs. Mclust</h1>
<h4 class="author"><em>Joyce Hsiao</em></h4>

</div>


<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-01-31</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> 8ea93f0</p>
<hr />
<div id="overviewresults" class="section level2">
<h2>Overview/Results</h2>
<p>I fit PAM and norma-based mixture models on the data that have already been filtered for quality single cells using both RNA-sequencing and microscopy data. The goal here is to identify a subset of cells that are less noisy and use these cells to estimate cell cycle phase.</p>
<p><span class="math inline">\(~\)</span></p>
<p>Results:</p>
<ol style="list-style-type: decimal">
<li><p>Compare PAM vs. Mclust results: seems that PAM places cluster centers at the most densely distributed region. and on the other hand, Mclust places cluster centers at the region that is the “center” of the points assigned to the clusters.</p></li>
<li><p>Select subset of cells: using PAM results, I compute silhouette index for all samples and the choose the top 25 samples in each cluster for each individual. These samples are used in some analysis to evaluate model fit on less noisy data, such as the analysis applying cellcycleR to both imaging data and sequencing data (<a href="https://jdblischak.github.io/fucci-seq/cellcycler-images-seqdata.html">here</a>).</p></li>
</ol>
<p>Let <span class="math inline">\(s(i)\)</span> denotes the silhouette index of sample <span class="math inline">\(i\)</span>. <span class="math inline">\(s(i)\)</span> ranges between -1 to 1. A large value of <span class="math inline">\(s(i)\)</span> indicates that sample <span class="math inline">\(i\)</span> is more similar to samples belonged to its own cluster than any other clusters, and a small value of <span class="math inline">\(s(i)\)</span> indicates that sample <span class="math inline">\(i\)</span> is dissimilar to samples belonged to its own clusters and matches better to samples belonged in some other clusters.</p>
<p><span class="math inline">\(a(i)\)</span>: Average distance of <span class="math inline">\(i\)</span> with all other samples within the same cluster.</p>
<p><span class="math inline">\(b(i)\)</span>: Lowest average distance of <span class="math inline">\(i\)</span> to all samples in any other cluster, of which <span class="math inline">\(i\)</span> is not a member. In other words, the average distance of sample <span class="math inline">\(i\)</span> to all samples in the neighboring cluster.</p>
<p><span class="math display">\[
s(i) = \frac{b(i)-a(i)}{max\{ a(i), b(i)\}}
\]</span></p>
<hr />
</div>
<div id="data-and-packages" class="section level2">
<h2>Data and packages</h2>
<p>Packages</p>
<pre class="r"><code>library(Biobase)
library(ggplot2)
library(cowplot)
library(cluster)
library(mclust)
library(data.table)
library(tidyr)</code></pre>
<p>Load data</p>
<pre class="r"><code>df &lt;- readRDS(file=&quot;../data/eset-filtered.rds&quot;)
pdata &lt;- pData(df)
fdata &lt;- fData(df)

# select endogeneous genes
counts &lt;- exprs(df)[grep(&quot;ERCC&quot;, rownames(df), invert=TRUE), ]</code></pre>
<hr />
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
<p>Fitting PAM for cells using only green and red intensity measurements.</p>
<pre class="r"><code>ints &lt;- with(pdata, data.frame(rfp.median.log10sum,
                               gfp.median.log10sum))
ints &lt;- data.frame(ints, 
                   chip_id = as.factor(pdata$chip_id))

k=3

pam_fit &lt;- lapply(1:uniqueN(ints$chip_id), function(i) {
    df_sub &lt;- subset(ints, chip_id==unique(chip_id)[i], 
                     select = -c(chip_id))
    fit_sub &lt;- pam(df_sub, k=k, diss=F)
    return(fit_sub)
  })
names(pam_fit) &lt;- unique(ints$chip_id)

mclust_fit &lt;- lapply(1:uniqueN(ints$chip_id), function(i) {
    df_sub &lt;- subset(ints, chip_id==unique(chip_id)[i], 
                     select = -c(chip_id))
    fit_sub &lt;- Mclust(df_sub, G=k)
    return(fit_sub)
})
names(mclust_fit) &lt;- unique(ints$chip_id)


# get centers of each cluster
pam_centers &lt;- do.call(rbind, lapply(1:uniqueN(ints$chip_id), function(i) {
  fit &lt;- pam_fit[[i]]
  ints_sub &lt;- ints[ints$chip_id == unique(ints$chip_id)[i],]
  tmp &lt;- fit$id.med
  tmp &lt;- data.frame(ints_sub[fit$id.med,],
                    center=c(1:3))
  return(tmp) }) )

mclust_centers &lt;- do.call(rbind, lapply(1:uniqueN(ints$chip_id), function(i) {
  fit &lt;- mclust_fit[[i]]
  tmp &lt;- fit$parameters$mean
  tmp &lt;- t(tmp)
  tmp &lt;- data.frame(tmp,center=c(1:3),
                    chip_id=names(mclust_fit)[i])
  return(tmp) }) )

# make data read for ggplot
pam_fit_plot &lt;- do.call(rbind, lapply(1:uniqueN(ints$chip_id), function(i) {
    df_sub &lt;- subset(ints, chip_id==unique(chip_id)[i])
    df_sub &lt;- data.frame(df_sub, cluster=as.factor(pam_fit[[i]]$clustering))
    return(df_sub)
}))

mclust_fit_plot &lt;- do.call(rbind, lapply(1:uniqueN(ints$chip_id), function(i) {
    df_sub &lt;- subset(ints, chip_id==unique(chip_id)[i])
    df_sub &lt;- data.frame(df_sub, cluster=as.factor(mclust_fit[[i]]$classification))
    return(df_sub)
}))</code></pre>
<hr />
</div>
<div id="cluster-centers" class="section level2">
<h2>Cluster centers</h2>
<p>Compare PAM vs. Mclust results</p>
<p><strong>k=3</strong></p>
<p><img src="figure/images-subset-silhouette.Rmd/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;" /></p>
<hr />
</div>
<div id="compute-silhouette" class="section level2">
<h2>Compute silhouette</h2>
<p>Combine silhouette data with the intensity data.</p>
<pre class="r"><code>si_pam &lt;- vector(&quot;list&quot;, length(pam_fit))
for (i in 1:length(si_pam)) {
  si_tmp &lt;- silhouette(pam_fit[[i]])

  si_out &lt;- lapply(1:3, function(cl) {
    ord &lt;- order(si_tmp[si_tmp[,1]==cl, 3], decreasing=TRUE)
    ii &lt;- as.numeric(rownames(si_tmp)[si_tmp[,1]==cl])
    ii &lt;- ii[ord]  
    df_sub &lt;- data.frame(ints[ii,],
                         unique_id = rownames(pdata)[ii],
                         cluster=cl,
                         si=si_tmp[si_tmp[,1]==cl, 3])
  }) 
  si_out &lt;- do.call(rbind, si_out)
  si_pam[[i]] &lt;- si_out
}
si_pam_long &lt;- do.call(rbind, si_pam)</code></pre>
<p>Choose the top 25 in each cluster.</p>
<pre class="r"><code>cutoff_nsamples &lt;- 20

foo &lt;- lapply(1:uniqueN(si_pam_long$chip_id), function(i) {
  foo2 &lt;- lapply(1:uniqueN(si_pam_long$cluster), function(j) {
    df_tmp &lt;- subset(si_pam_long, chip_id == unique(si_pam_long$chip_id)[i] &amp; cluster == unique(si_pam_long$cluster)[j])
    if (cutoff_nsamples &gt; dim(df_tmp)[1]) {
        df_tmp_sub &lt;- df_tmp
    } else {
        df_tmp_sub &lt;- df_tmp[which(order(df_tmp$si, decreasing = T) %in% c(1:cutoff_nsamples)),]
    }
    return(df_tmp_sub)
  })
  foo2 &lt;- do.call(rbind, foo2)
  return(foo2)
})
foo &lt;- do.call(rbind, foo)
si_pam_25 &lt;- foo</code></pre>
<p>Compare full set versus subset of cells.</p>
<p><img src="figure/images-subset-silhouette.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>hard code cluster labels. default in ggplot is 1 orange, 2 green, 3 blue</p>
<pre class="r"><code># NA18511
# orange 1 to 3, green 2 to 2, blue 3 to 1, 

# NA18855
# orange 1 to 2, green 2 to 3, blue 3 to 1, 

# NA18870
# orange 1 to 2, green 2 to 1, blue 3 to 3

# NA19098
# orange 1 to 1, green 2 to 3, blue 3 to 2

# NA19101
# orange 1 to 3, green 2 to 2, blue 3 to 1

# NA19160
# orange 1 to 2, green 2 to 3, blue 3 to 1

tmp &lt;- si_pam_long
tmp[si_pam_long$chip_id == &quot;NA18511&quot; &amp; si_pam_long$cluster==1,]$cluster &lt;- 3
tmp[si_pam_long$chip_id == &quot;NA18511&quot; &amp; si_pam_long$cluster==2,]$cluster &lt;- 2
tmp[si_pam_long$chip_id == &quot;NA18511&quot; &amp; si_pam_long$cluster==3,]$cluster &lt;- 1

tmp[si_pam_long$chip_id == &quot;NA18855&quot; &amp; si_pam_long$cluster==1,]$cluster &lt;- 2
tmp[si_pam_long$chip_id == &quot;NA18855&quot; &amp; si_pam_long$cluster==2,]$cluster &lt;- 3
tmp[si_pam_long$chip_id == &quot;NA18855&quot; &amp; si_pam_long$cluster==3,]$cluster &lt;- 1

tmp[si_pam_long$chip_id == &quot;NA18870&quot; &amp; si_pam_long$cluster==1,]$cluster &lt;- 2
tmp[si_pam_long$chip_id == &quot;NA18870&quot; &amp; si_pam_long$cluster==2,]$cluster &lt;- 1
tmp[si_pam_long$chip_id == &quot;NA18870&quot; &amp; si_pam_long$cluster==3,]$cluster &lt;- 3

tmp[si_pam_long$chip_id == &quot;NA19098&quot; &amp; si_pam_long$cluster==1,]$cluster &lt;- 1
tmp[si_pam_long$chip_id == &quot;NA19098&quot; &amp; si_pam_long$cluster==2,]$cluster &lt;- 3
tmp[si_pam_long$chip_id == &quot;NA19098&quot; &amp; si_pam_long$cluster==3,]$cluster &lt;- 2

tmp[si_pam_long$chip_id == &quot;NA19101&quot; &amp; si_pam_long$cluster==1,]$cluster &lt;- 3
tmp[si_pam_long$chip_id == &quot;NA19101&quot; &amp; si_pam_long$cluster==2,]$cluster &lt;- 2
tmp[si_pam_long$chip_id == &quot;NA19101&quot; &amp; si_pam_long$cluster==3,]$cluster &lt;- 1

tmp[si_pam_long$chip_id == &quot;NA19160&quot; &amp; si_pam_long$cluster==1,]$cluster &lt;- 2
tmp[si_pam_long$chip_id == &quot;NA19160&quot; &amp; si_pam_long$cluster==2,]$cluster &lt;- 3
tmp[si_pam_long$chip_id == &quot;NA19160&quot; &amp; si_pam_long$cluster==3,]$cluster &lt;- 1


plot_grid(
  ggplot(data=si_pam_long) + 
      geom_point(aes(x=gfp.median.log10sum, 
                     y=rfp.median.log10sum, col=as.factor(cluster)),
         alpha = .5, cex = .7) + 
      geom_point(data=pam_centers,
                  aes(x=gfp.median.log10sum,
                      y=rfp.median.log10sum), shape=4, size=3) +
      labs(title = &quot;K=3, PAM&quot;,
           x=&quot;GFP intensity (log10 pixel sum)&quot;, 
           y = &quot;RFP intensity (log10 pixel sum)&quot;) + 
      facet_wrap(~as.factor(chip_id), ncol=3) +
      theme_gray() + theme(legend.position=&quot;none&quot;),
  ggplot(data=si_pam_25) + 
      geom_point(aes(x=gfp.median.log10sum, 
                     y=rfp.median.log10sum, col=as.factor(cluster)),
         alpha = .5, cex = .7) + 
      geom_point(data=pam_centers,
                  aes(x=gfp.median.log10sum,
                      y=rfp.median.log10sum), shape=4, size=3) +
      labs(title = &quot;Top 25 samples within clusters&quot;,
           x=&quot;GFP intensity (log10 pixel sum)&quot;, 
           y = &quot;RFP intensity (log10 pixel sum)&quot;) + 
      facet_wrap(~as.factor(chip_id), ncol=3) +
      theme_gray() + theme(legend.position=&quot;none&quot;)
  )</code></pre>
<p><img src="figure/images-subset-silhouette.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>save subset of cells to rds.</p>
<pre class="r"><code>save(si_pam_long, si_pam_25, 
    file = &quot;../output/images-subset-silhouette.Rmd/si_pam.rda&quot;)</code></pre>
<hr />
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre><code>R version 3.4.1 (2017-06-30)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS: /home/joycehsiao/miniconda3/envs/fucci-seq/lib/R/lib/libRblas.so
LAPACK: /home/joycehsiao/miniconda3/envs/fucci-seq/lib/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] tidyr_0.7.1         data.table_1.10.4   mclust_5.3         
[4] cluster_2.0.6       cowplot_0.9.1       ggplot2_2.2.1      
[7] Biobase_2.38.0      BiocGenerics_0.24.0

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.14     knitr_1.17       magrittr_1.5     munsell_0.4.3   
 [5] colorspace_1.3-2 rlang_0.1.4.9000 stringr_1.2.0    plyr_1.8.4      
 [9] tools_3.4.1      grid_3.4.1       gtable_0.2.0     git2r_0.19.0    
[13] htmltools_0.3.6  yaml_2.1.16      lazyeval_0.2.0   rprojroot_1.2   
[17] digest_0.6.12    tibble_1.3.3     purrr_0.2.2      glue_1.1.1      
[21] evaluate_0.10.1  rmarkdown_1.8    labeling_0.3     stringi_1.1.2   
[25] compiler_3.4.1   scales_0.4.1     backports_1.0.5 </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
