<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">fucci-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/fucci-seq">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>We used this site to collaborate and share our results. Please feel free to explore. The results that made it into the final paper are in the section <a href="#finalizing">Finalizing</a> below. Here are some useful links:</p>
<ul>
<li><a href="doi.org/10.1101/526848">Characterizing and inferring quantitative cell-cycle phase in single-cell RNA-seq data analysis</a></li>
<li>GEO record <a href="http://www.ncbi.nlm.nih.gov/%20geo/query/acc.cgi?acc=GSE121265">GSE121265</a> for the raw FASTQ files</li>
<li><a href="https://github.com/jdblischak/fucci-seq">GitHub repo</a> for code and data</li>
<li><a href="https://github.com/jhsiao999/peco">peco package</a> for our supervised approach predicting cell cycle phase in induced pluripotent stem cells.</li>
</ul>
<div id="finalizing" class="section level1">
<h1>Finalizing</h1>
<div id="data-description" class="section level3">
<h3><a href="data-overview.html">Data description</a></h3>
</div>
<div id="quality-control-of-scrna-seq-data" class="section level3">
<h3><a href="#rna-seq-data-preprcessing">Quality control of scRNA-seq data</a></h3>
</div>
<div id="process-images---from-images-to-intensities" class="section level3">
<h3><a href="images-process.html">Process images - from images to intensities</a></h3>
</div>
<div id="inspect-and-correct-for-the-c1-plate-effect-in-fluorescence-intensities" class="section level3">
<h3><a href="images-normalize-anova.html">Inspect and correct for the C1 plate effect in fluorescence intensities</a></h3>
</div>
<div id="infer-an-angel-for-each-cell-based-on-fucci-fluorescence-intensities" class="section level3">
<h3><a href="images-circle-ordering-eval.html">Infer an angel for each cell based on FUCCI fluorescence intensities</a></h3>
</div>
<div id="estimate-the-cyclic-trends-of-gene-expression-levels" class="section level3">
<h3><a href="npreg-trendfilter-quantile.html">Estimate the cyclic trends of gene expression levels</a></h3>
<!-- * Code for compiling results and plots for the paper: `code/makeresults_paper.R` -->
<!-- ## Supervised method learning: finalizing results on holding-an-individual-out scenario -->
<!--     * [Validation sample](method-eval.html) (to be removed...) -->
<!-- * Predicting individual X using expression from random individuals -->
<!--     * [Selecting the top X cyclical genes in the training sample](method-train-ind.html) -->
<!--     * [Learning gene annotations](method-train-ind-genes.html) -->
<!--     * [Validation sample](method-eval-ind.html) (to be removed) -->
<!-- * Analyze the public datasets, including [Leng et al.](method-eval-leng.html), [Bottcher et al.](method-eval-bottcher.html), and [Buettner et al.](method-eval-buettner.html). -->
</div>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<div id="supervised-method-learning-finalizing-on-data-holiding-out-validation-samples" class="section level2">
<h2>Supervised method learning: finalizing on data holiding out validation samples</h2>
<ul>
<li>Perform model training on cell times computed in different ways
<ul>
<li>Based on Fucci only
<ul>
<li>Cells from mixed individuals: <a href="method-train-classifiers-all.html">Prediction error</a>, and <a href="method-train-classifiers-genes.html">functional annotations of genes selected</a></li>
<li>Cells from one individual: <a href="method-train-ind.html">Prediction error</a>, and <a href="method-train-ind-genes.html">functional annotations of genes selected</a></li>
</ul></li>
<li>Based on Fucci and DAPI and derived from trigonometric transformation
<ul>
<li>Cells from mixed individuals: <a href="method-train-triple.html">Prediction error</a></li>
<li>Cells from one individual: <a href="method-train-ind.html">Prediction error</a>, and <a href="method-train-ind-genes.html">functional annotations of genes selected</a></li>
</ul></li>
<li>Based on FUCCI and DAPI and derived from an algebraic approach
<ul>
<li><a href="method-train-alge.html">Cells from one individual</a></li>
</ul></li>
</ul></li>
<li>Compare model training results
<ul>
<li><a href="method-labels-compare.html">Compute and compare cell time estimates</a> derived under two different assumptions: assuming equal variance in PC1 and PC2 (circle) vs. unequal variances in PC1 and PC2.</li>
<li><a href="method-train-genes-fucci-vs-triple.html">Prediction error for cells from mixed individuals</a>: Prediction error smaller when based on fucci time only than based on fucci and dapi time, though only a very small difference. However, the genes selected in these two are similar, both found to have 37 genes enriched for the Cell Cycle GO (0007049).</li>
</ul></li>
<li>Compiling results
<ul>
<li><a href="method-train-summary-output.html">Compile training results for both training scenarios</a> and <a href="method-train-summary.html">Plotting prediction error margin and genes selected</a></li>
<li><a href="method-validation.html">Compile results on validation samples</a></li>
</ul></li>
</ul>
</div>
<div id="supervised-method-learning-building-the-analysis-approach" class="section level2">
<h2>Supervised method learning: building the analysis approach</h2>
<ul>
<li><p>Early analysis <a href="method-npreg.html">using trendfilter and evaluating model performance</a></p></li>
<li><p>Investigate the property and quality of <a href="method-labels.html">the cell time labels derived from GFP and RFP</a>: PVE of intensities by cell times, comparing PVEs by cell times from DAPI and FUCCI vs from FUCCI; comparing prediction before and after removing PC outliers (after is slightly better), and in different sets of genes.</p></li>
<li><p>Investigate the cyclical trends for <a href="trendfilter-individua.Rmd">each individual</a></p></li>
<li><p>Investigate the property of <a href="method-label-seurat.html">Seurat classes and cell time</a></p></li>
<li>Noisy label analysis
<ul>
<li>PVE in DAPI, GFP, and RFP <a href="method-labels-noisy.html">before vs after removing noisy labels</a></li>
<li>Comparing results excluding/including noisy training labels on top 10 and top 101 cyclical genes, when <a href="method-train-labels.html">excluding noisy labels from each fold validation, hence unequal sample numbers per fold</a> as opposed to <a href="method-train-labels-equalfold.html">excluding noisy labels before making fold valdiation, hence equal nubmer per fold</a></li>
</ul></li>
<li>Analyze the training datasets
<ul>
<li>Comparing trendfilter order 2 and order 3 with npcirc.nw <a href="method-npreg-prelim-results.html">on the odd-numbered genes in the top 101 cyclical genes</a></li>
<li>Comparing resuls of the various classifers (supervised, PC, unsupervised) <a href="method-train-classifiers.html">on top 101 cyclical genes</a>, and on <a href="method-train-classifiers-top10.html">top 10 cyclial genes</a>.</li>
</ul></li>
<li>Analyze the withheld samples
<ul>
<li>Evaluate the withheld sample <a href="method-eval-withheld-explore.html">including noisy labels</a> on the top 10 and top 101 cyclical genes] and comparing peco methods with seurat predictions.</li>
<li>Evaluate the withheld samples <a href="method-eval-withheld-explore-removenoisy.html">removing noisy labels</a> on the top 10 and top 101 cyclical genes and comparing peco methods with seurat predictions.</li>
</ul></li>
</ul>
</div>
<div id="approaches-to-fitting-cyclical-trend-in-gene-expression-data" class="section level2">
<h2>Approaches to fitting cyclical trend in gene expression data</h2>
<ul>
<li>Apply spherically projected multivariate linear model
<ul>
<li><a href="model-spml.pilot.html">First trying out spml package</a></li>
<li><a href="method-spml.html">To predictd cell times using traing/test samples set-up</a></li>
</ul></li>
<li><p>Learning about correlation for circular response variable: some simulations to check <a href="circ-simulation-correlation.html">its property and also relations with Fisher’s z transformation</a></p></li>
<li>CellcycleR 0.1.6
<ul>
<li><a href="images-cellcycleR-convergence.html">Model convergence assessment</a></li>
<li><a href="images-cellcycleR.html">Fitting on intensities across plates and individuals</a></li>
<li><a href="cellcycler-seqdata-leng.html">Fitting on Leng data)</a></li>
<li><a href="cellcycler-seqdata-fucci.html">Fitting on fucci-seq RNA-seq data)</a></li>
</ul></li>
</ul>
</div>
<div id="cell-cycle-signal-in-gene-expression-data" class="section level2">
<h2>Cell cycle signal in gene expression data</h2>
<ol style="list-style-type: decimal">
<li>We investigated cell cycle signals in the sequencing data alone.
<ul>
<li>Consider <a href="images-transgene.html">transgene count in sequencing data</a></li>
<li>Compute linear correlation between gene expression levels <a href="images-seq-correlation.html">with DAPI and FUCCI intensities</a></li>
</ul></li>
<li>We then assign categorical labels of cell cycle and explored the expresson profiles of these categories.
<ul>
<li>Cluster samples by <a href="images-pam.html">Partition around medoids(PAM)</a></li>
<li>Cluster samples by <a href="images-mclust.html">Guassian mixture modeling</a></li>
<li>Select a subset of samples that are closet to the cluster centers (cluster representives) <a href="images-subset-silhouette.html">using silhouette index</a></li>
<li>Examine gene pression scores defined by the Macosko paper in the selected cluster representatives <a href="images-classify-fucci.html">before confounding correction</a> and <a href="images-classify-fucci-adjusted.html">after confounding correction</a></li>
<li>Examine gene expression scores defined by the Macosk paer <a href="images-classify-leng.html">in the sorted cells of the Leng et al. 2015 paper</a></li>
</ul></li>
<li>We inferred an angle for each cell on a unit circle using FUCCI intensities alone.
<ul>
<li>First, I used GFP and RFP intensities to estimate a <a href="images-circle-ordering.html">least-square fit of unit circle</a></li>
<li>I computed the PCs of GFP and RFP and used these to infer an angle for each cell on a unit circle (i.e., polar coordinates), and <a href="images-circle-ordering-eval.html">to evaluate these results, I computed circular-circular and circular-linear correlation with DAPI and gene expression</a></li>
<li>Here I put together lists of genes identified as significantly correlatd with the PC-based fit <a href="images-circle-ordering-sigcorgenes.html">by linear correlation and circular-linear correlation, also cyclical genes by smash</a></li>
</ul></li>
<li>I used nonparametric methods to identify genes that may be cyclical along cell cycle phases.
<ul>
<li>Fit <a href="npreg-methods.html">smash and kernel regression on circular variables</a> on a subset of genes with detection rate &gt; .8.</li>
<li>Fit <a href="npreg-trendfilter-prelim.html">trendfilter</a> on a subset of genes (5) that are observed (visually) to have cyclical pattern. trendfilter is robust to small proportion of undetected cells, approx 2 or 3%. In cases of simulation when increasing proportion of undetected cells to 20%, we observed a flat line in gene expression for genes previously identified to tend to a cyclical pattern.</li>
<li>Next, we fit trendfilter on all genes after transforming the data to follow standard normal distribution, permutation-based p-values for PVE are used to select <a href="npreg-trendfilter-quantile.html">101 significant cyclical genes</a>.</li>
</ul></li>
<li>Additional analysis done to identify top cyclical genes in each individual. The top 5 are not shared across the six individuals. <a href="trendfilter-individual.html">Results</a></li>
</ol>
</div>
<div id="rna-seq-data-preprcessing" class="section level2">
<h2>RNA-seq data preprcessing</h2>
<ol style="list-style-type: decimal">
<li>The first step in preprocessing RNA-seq data consists of QC and filtering.
<ul>
<li>Sample QC and filtering
<ul>
<li><a href="sampleqc.html">Sample QC criteria</a><br />
</li>
<li><a href="totals.html">Sequencing depth</a><br />
</li>
<li><a href="reads-v-molecules.html">Reads versus molecules</a></li>
</ul></li>
<li>Gene QC and filtering
<ul>
<li><a href="gene-filtering.html">gene filtering</a></li>
<li><a href="pca-tf.html">PCA with technical fators</a></li>
</ul></li>
</ul></li>
<li>We then analyzed and corrected for batch effect due to C1 plate in the sequencing data
<ul>
<li><a href="seqdata-batch-correction.html">Estimate variance explained in IBD and correct for batch effects</a></li>
</ul></li>
</ol>
</div>
<div id="microscopy-image-analysis" class="section level2">
<h2>Microscopy image analysis</h2>
<ul>
<li><a href="images-process.html">Processing images - from images to intensities</a></li>
</ul>
<p>We evaluated and pre-processed the results of image analysis as follows:</p>
<ol style="list-style-type: decimal">
<li>We visually inspect images deteced to have none or more than one nucleus. For cases that are inconsistent with visual inspection, we correct the number of nuclei detected.
<ul>
<li><a href="images-multiple-nuclei.html">Inspect images with multiple nuclei</a></li>
<li><a href="images-zero-nuclei.html">Inspect images with no nucleus</a></li>
</ul></li>
<li>We applied background correction to the intensity measurements of GFP, RFP and DAPI based on the following analyses.
<ul>
<li><a href="confess-prelim.html">CONFESS results</a></li>
<li><a href="images-qc.html">QC analysis including no. nuclei detected, DAPI, and intensity variation</a></li>
<li><a href="images-qc-followup.html">Explore using log10 sum pixel intensity for signal metrics</a></li>
<li><a href="images-metrics.html">Compare correction approaches using median versus mean background</a></li>
<li><a href="images-metrics-cell-shape.html">Explore associations between nucleus shape metrics vs intensities</a></li>
</ul></li>
<li>We analyzed intensity variation across individuals and batches and considers approaches for removing batch effects in the data.
<ul>
<li><a href="images-qc-labels.html">Visualize signal variation by plate and individual identity</a></li>
<li><a href="images-qc-variation.html">Visualize the structure of signal variation by individual identity</a></li>
<li><a href="images-normalize-quantile.html">Quantile normalization for GFP, RFP and DAPI</a></li>
<li><a href="images-normalize-anova.html">Estimate variance explained in IBD and correct for batch effects in intensities</a></li>
</ul></li>
<li>We investigated the cell time estimates based on FUCCI intensities.
<ul>
<li>Consider the mean-adjusted FUCCI intensities, what’s the relationship between cell time <a href="images-time-eval.html">estimates and DAPI and sample molecule count</a>?</li>
<li>Consider the quantile-normalized FUCCI intensities versus the mean-adjusted intensities (for adjusting batch effect). What’s the differences between cell time estimates from <a href="images-time-compare-normalize.html">the two normalization approaches</a>?</li>
</ul></li>
</ol>
</div>
<div id="one-time-investigations" class="section level2">
<h2>One-time investigations</h2>
<ul>
<li><p>Why some gene symbols (genes) correspond to <a href="ensembl.html">multiple Ensembl IDs?</a></p></li>
<li><p>I selected a set of cell cycle <a href="seqdata-select-cellcyclegenes.html">genes that belong to GO term Cell Cycle and looked at the overlap with the detected genes in our data.</a></p></li>
<li><p><a href="seurat-cellcycle.Rmd">Replicate Seurat example of scoring cell cycle phases</a></p></li>
<li><p><a href="talk-20180416.html">Reproducing some QC analysis and making plots for a seminar talk</a></p></li>
<li><p>[Compile codes for the paper in ]</p></li>
</ul>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
