---
title: "Data overview"
author: "Joyce Hsiao"
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

---

## Quick start

We store feature-level (gene) read count and molecule count in `expressionSet` (`data/eset`) objects, which also contain sample metadata (e.g., assigned indivdual ID, cDNA concentraion) and quality filtering criteria (e.g., number of reads mapped to FUCCI transgenes, ERCC conversion rate). Data from different C1 plates are stored in separate `eset` objects:

$~$

To combine `eset` objects from the different C1 plates:

`eset <- Reduce(combine, Map(readRDS, Sys.glob("../data/eset/*.rds")))`

$~$

To access data stored in  `expressionSet`:

* `exprs(eset)`: access count data, 20,421 features by 1,536 single cell samples.

* `pData(eset)`: access sample metadata. Returns data.frame of 1,536 samples by 43 labels. Use `varMetadata(phenoData(eset))` to view label descriptions. 

* `fData(eset)`: access feature metadata. Returns data.frame of 20,421 features by 6 labels. Use `varMetadata(featureData(eset))` to view label descriptions. 


---

## Label description

Load related packages.

```{r, message=FALSE}
library(knitr)
library(Biobase)
```

To combine all `expressionSet` objects in the folder, 

```{r}
fname <- Sys.glob("../data/eset/*.rds")
eset <- Reduce(combine, Map(readRDS, fname))
```

View the sample metadata labels,

```
varMetadata(phenoData(eset))
```

```{r, eval = TRUE, echo = FALSE}
kable(varMetadata(phenoData(eset)))
```

View the feature (gene) metadata labels,

```
varMetadata(featureData(eset))
```

```{r, eval=TRUE, echo = FALSE}
kable(varMetadata(featureData(eset)))
```

---

## Session information

```{r, echo = FALSE}
sessionInfo()
```

