---
title: "Gene filtering"
author: "Joyce Hsiao"
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

---

## Summary

I performed gene filtering based on the criterion set forth in our previous paper. 

1. Remove mitochrodrial genes: filter out mitochrondrial genes verified and listed in [MitoCarta][mitocarta].

*Results*:Found 1,150 genes previously quantified in `MitoCarta` inventory.

*Output*: gene annotation saved in `../output/gene-filtering.Rmd/mito-genes-info.csv`

$~$

2. Remove outlier genes: molecule counts > 4,096 in any sample (x is the theoretical maximum of UMI count for 6-bp UMI)

*Results* There's one, and turns out this over-expressed gene is one of the mitochrondrial genes.

*Output*: gene annotation saved in `../output/gene-filtering.Rmd/over-expressed-genes-info.csv`

$~$

3. Remove lowly expressed genes: Lowly-expressed genes := gene mean < 2 CPM.

*Results*: * Of 20,421 genes, 7,864 genes are classifed as lowly-expressed. Of these, 34 are ERCC control genes and 7,830 are endogenoeus genes. 

*Output*: gene annotation saved in `../output/gene-filtering.Rmd/lowly-expressed-genes-info.csv`


**Finally**, filtered eset (`eset_filtered`) and cpm normalized count (`cpm_filtered`) are saved in `../output/gene-filtering.Rmd/eset-filterd.rdata`.

[mitocarta]: https://www.broadinstitute.org/scientific-community/science/programs/metabolic-disease-program/publications/mitocarta/mitocarta-in-0


$~$

---

## Import data

Combine `eSet` objects.

```{r, message=FALSE}
library(knitr)
library(Biobase)
#library(gdata)
library(testit)
library(cowplot)
library(biomaRt)
library(knitr)
library(data.table)

source("../code/pca.R")
eset <- readRDS("../output_tmp/eset.rds")
```

Filter out low-quality single cell samples.

```{r}
pdata_filter <- pData(eset)[pData(eset)$filter_all == TRUE,]
count_filter <- exprs(eset[,pData(eset)$filter_all == TRUE])
dim(count_filter)
```

$~$

---

## Mitochrondrial genes

Found 1,150 genes previously quantified in `MitoCarta` inventory.

```{r, warning = FALSE}
human_mito <- gdata::read.xls("../data/Human.MitoCarta2.0.xls",
                   sheet = 2, header = TRUE, stringsAsFactors=FALSE)
human_mito_ensembl <- human_mito$EnsemblGeneID

which_mito <- which(rownames(count_filter) %in% human_mito_ensembl)
which_mito_genes <- rownames(count_filter)[which_mito]

length(which_mito)
```

Get mito gene info via `biomaRt`.

```{r, eval = F}
# do biomart to verfiy these genes
ensembl <- useMart(host = "grch37.ensembl.org",
                   biomart = "ENSEMBL_MART_ENSEMBL",
                   dataset = "hsapiens_gene_ensembl")

mito_genes_info <- getBM(
  attributes = c("ensembl_gene_id", "chromosome_name",
                 "external_gene_name", "transcript_count",
                 "description"),
  filters = "ensembl_gene_id",
  values = which_mito_genes[grep("ENSG", which_mito_genes)],
  mart = ensembl)


fwrite(mito_genes_info, 
       file = "../output/gene-filtering.Rmd/mito-genes-info.csv")
```

$~$

---

## Over-expressed genes

There's one, and turns out this over-expressed gene is one of the mitochrondrial genes.

```{r}
which_over_expressed <- which(apply(count_filter, 1, function(x) any(x>(4^6)) ))
over_expressed_genes <- rownames(count_filter)[which_over_expressed]
over_expressed_genes %in% human_mito_ensembl
over_expressed_genes
```

Get over-expressed gene info via `biomaRt`.

```{r, eval = FALSE}
over_expressed_genes_info <- getBM(
  attributes = c("ensembl_gene_id", "chromosome_name",
                 "external_gene_name", "transcript_count",
                 "description"),
  filters = "ensembl_gene_id",
  values = over_expressed_genes,
  mart = ensembl)

fwrite(over_expressed_genes_info, 
       file = "../output/gene-filtering.Rmd/over-expressed-genes-info.csv")
```

$~$

---

## Filter out lowly-expressed genes

* Of 20,421 genes, 7,864 genes are classifed as lowly-expressed. Of these, 34 are ERCC control genes and 7,830 are endogenoeus genes. 

Compute CPM

```{r}
cpm <- t(t(count_filter)/colSums(count_filter))*(10^6)
```

Lowly-expressed genes := gene mean < 2 CPM

```{r}
which_lowly_expressed <- which(rowMeans(cpm) < 2)
length(which_lowly_expressed)

which_lowly_expressed_genes <- rownames(cpm)[which_lowly_expressed]

length(grep("ERCC", which_lowly_expressed_genes))
length(grep("ENSG", which_lowly_expressed_genes))
```

Get gene info via `biomaRt`.

```{r, eval = FALSE}
lowly_expressed_genes_info <- getBM(
  attributes = c("ensembl_gene_id", "chromosome_name",
                 "external_gene_name", "transcript_count",
                 "description"),
  filters = "ensembl_gene_id",
  values = which_lowly_expressed_genes[grep("ENSG", which_lowly_expressed_genes)],
  mart = ensembl)

fwrite(lowly_expressed_genes_info, 
       file = "../output/gene-filtering.Rmd/lowly-expressed-genes-info.csv")
```

$~$

---

## Combine filters

Including 16,460 genes.

```{r}
gene_filter <- unique(c(which_lowly_expressed, which_mito, which_over_expressed))

genes_to_include <- setdiff(1:nrow(count_filter), gene_filter)
length(genes_to_include)
```

$~$

---

## Make filtered data

```{r}
cpm_filtered <- cpm[genes_to_include, ]

eset_filtered <- eset[genes_to_include, pData(eset)$filter_all==TRUE]
eset_filtered
```

```{r, echo = TRUE, eval=F}
save(cpm_filtered, eset_filtered, 
     file = "../output/gene-filtering.Rmd/eset-filtered.rdata")
```

$~$

---

## Compute log2 CPM

Import data post sample and gene filtering.

```{r}
load(file="../output/gene-filtering.Rmd/eset-filtered.rdata")
```

Compute log2 CPM based on the library size before filtering.

```{r}
log2cpm <- log2(cpm_filtered+1)
```

$~$

---

## PCA

```{r}
pca_log2cpm <- run_pca(log2cpm)

pdata <- pData(eset_filtered)
pdata$experiment <- as.factor(pdata$experiment)

plot_grid(
  plot_pca(x=pca_log2cpm$PCs, explained=pca_log2cpm$explained,
         metadata=pdata, color="chip_id"),
  plot_pca(x=pca_log2cpm$PCs, explained=pca_log2cpm$explained,
         metadata=pdata, color="experiment"),
  ncol=2)
```

$~$

---

## Session information

```{r, echo = FALSE}
sessionInfo()
```

