---
title: "Expression profile of the gold-standard samples"
author: "Joyce Hsiao"
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning=FALSE)
```

---

## Overview/Results

1. The goal here is to assess the expression profiles of samples within the assigned clusters based on FUCCI measures ([see here for details](https://jdblischak.github.io/fucci-seq/images-subset-silhouette.html)).

2. Results show somewhat variable patterns across the assigned clusters (within each individual).

3. Note that the genes used for the analysis were previously identified as "variable" or "cell cycle regulating" genes in [Macosko et al. 2015](http://dx.doi.org/10.1016/j.cell.2015.05.002). Applying the 544 genes in their results, we found 230 genes identified as "variable" in expression profiles along cell cycle phases. 

4. For validity of the phase scores, we further apply the same method to data in [Leng et al. 2015](doi:10.1038/nmeth.3549), which have been previously scored for cell cycle phases also using FUCCI measurse. 

---

## Data and packages

Packages

```{r}
library(Biobase)
library(ggplot2)
library(cowplot)
library(data.table)
library(tidyr)
library(gplots)
```

Load data

```{r}
df <- readRDS(file="../data/eset-filtered.rds")
pdata <- pData(df)
fdata <- fData(df)

# select endogeneous genes
counts <- exprs(df)[grep("ERCC", rownames(df), invert=TRUE), ]

# cpm normalization
log2cpm <- log2(t(t(counts+1)*(10^6)/colSums(counts)))
```

select cell cycle genes

```{r}
cellcycle <- readRDS("../data/cellcycle-genes-previous-studies/rds/macosko-2017.rds")
which_cc <- which(rownames(log2cpm) %in% cellcycle$ensembl)

log2cpm_cc <- log2cpm[which_cc, ]
cc_genes <- cellcycle[which(cellcycle$ensembl %in% rownames(log2cpm_cc)),]
```

intensity data

```{r}
ints <- with(pdata, data.frame(rfp.median.log10sum,
                               gfp.median.log10sum,
                               chip_id))
```

Load best subset. `si_pam.rda` contains two objects: `si_pam_25` (top 25 samples within clusters and individuals) and `si_pam_long` (all sample information on silhouette index). 

```{r}
load(file = "../output/images-subset-silhouette.Rmd/si_pam.rda")
```

Combine sample phoenotye labels with the silhouette sample top 25. 

```{r}
si_pam_25$molecules <- pdata$molecules[match(si_pam_25$unique_id, rownames(pdata))]
```

---


## Analysis

Compute phase-specific score for within clusters and individuals. This method was previously used in [Macosko et al. 2015](http://dx.doi.org/10.1016/j.cell.2015.05.002) for identifying gene expression patterns that varied along cell cycle phases, and for summarizing cell cycle phase profile for single cell samples. 

We applied the 544 genes identified as varied in [Macosko et al. 2015](http://dx.doi.org/10.1016/j.cell.2015.05.002) and identified 230 genes that varied in expression patterns. 

The method was applied to log2CPM normalized data. Briefly,

1. Identify the variable genes wihtin each cell cycle phase: compute for each cell-cycle phase, correlation between per-gene expression level and mean gene expression levels across all single cell samples. Select genes with correlation > .3. 

2. Compute phase-specific score: compute average expression across genes for each single cell samples. 

3. Standardize phase-specific scores in two-steps: within each phase, standardize (transforming to z-scores) scores across single cell samples, and then within each single cell sample, standardize scores across phase. 


```{r}
cc_scores <- lapply(1:uniqueN(cc_genes$phase), function(i) {
  ph <- unique(cc_genes$phase)[i]
  df_sub <- log2cpm_cc[rownames(log2cpm_cc) %in% cc_genes$ensembl[cc_genes$phase == ph],]
  mn <- colMeans(df_sub)
  cc <- cor(t(rbind(mn, df_sub)))
  cc_mean <- cc[-1,1]
  genes_cc <- names(cc_mean)[which(cc_mean > .3)]
  scores_raw <- colMeans(df_sub[rownames(df_sub) %in% genes_cc,])
  scores_z <- scale(scores_raw)
  return(scores_z)
})
names(cc_scores) <- unique(cc_genes$phase)

cc_scores <- do.call(cbind, cc_scores)
colnames(cc_scores) <- unique(cc_genes$phase)

# standardize scores across phases
cc_scores_z <- t(apply(cc_scores, 1, scale))
colnames(cc_scores_z) <- unique(cc_genes$phase)
cc_scores_z <- as.data.frame(cc_scores_z)

# convert data format from wide to long
cc_scores_z_long <- gather(cc_scores_z, key=phase, value=cc_scores_z)
cc_scores_z_long$uniqe_id <- rep(rownames(cc_scores_z), ncol(cc_scores_z))

cc_scores_z_long$chip_id <- pdata$chip_id[match(cc_scores_z_long$uniqe_id, rownames(pdata))]
cc_scores_z_long$experiment <- pdata$experiment[match(cc_scores_z_long$uniqe_id, rownames(pdata))]


# select gold standard set
cc_scores_z_long$cluster <- si_pam_25$cluster[match(cc_scores_z_long$uniqe_id, si_pam_25$unique_id)]

cc_scores_z_long_25 <- cc_scores_z_long[!is.na(cc_scores_z_long$cluster),]

cc_scores_z_long_25$phase <- factor(cc_scores_z_long_25$phase, 
                                    levels=c("M/G1", "G1/S", "S", "G2", "M"))
cc_scores_z_long_25$cluster <- as.factor(cc_scores_z_long_25$cluster)
```


```{r, fig.width=9, fig.height=7, echo = F}
ggplot(data=cc_scores_z_long_25, 
       aes(x=phase, y=cc_scores_z, col=phase)) +
  geom_violin() +
  geom_boxplot(width=.1) +
  facet_grid(as.factor(chip_id)~cluster) +
  ylab("phase-specific scores") +
  ggtitle("scores within phases and individuals")
```

```{r, fig.width=9, fig.height=7, echo = F}
ggplot(data=cc_scores_z_long_25, 
       aes(x=cluster, y=cc_scores_z, col=cluster)) +
  geom_violin() +
  geom_boxplot(width=.1) +
  facet_grid(as.factor(chip_id)~phase) +
  ylab("phase-specific scores") +
  ggtitle("scores within cluster and individuals")
```

---

## Others

Look up "classical cell cycle genes" listed in Macosko et al. 2015 and see their patterns across the assigned clusters.

```{r}
hgnc <- c("CCNB1", "CCNB2", "MCM2", "MCM3", "MCM4", "MCM5", "MCM6",
          "MCM7", "MCM10", "AURKA", "AURKB")
ensg <- cc_genes$ensembl[which(cc_genes$hgnc %in% hgnc)]
tmp <- log2cpm_cc[which(rownames(log2cpm_cc) %in% ensg),]
tmp <- data.frame(t(tmp))

cc_scores_z_long_25$ENSG00000073111 <- tmp$ENSG00000073111[match(cc_scores_z_long_25$uniqe_id, rownames(tmp))]
cc_scores_z_long_25$ENSG00000076003 <- tmp$ENSG00000076003[match(cc_scores_z_long_25$uniqe_id, rownames(tmp))]
cc_scores_z_long_25$ENSG00000087586 <- tmp$ENSG00000087586[match(cc_scores_z_long_25$uniqe_id, rownames(tmp))]
cc_scores_z_long_25$ENSG00000076003 <- tmp$ENSG00000076003[match(cc_scores_z_long_25$uniqe_id, rownames(tmp))]
cc_scores_z_long_25$ENSG00000100297 <- tmp$ENSG00000100297[match(cc_scores_z_long_25$uniqe_id, rownames(tmp))]
cc_scores_z_long_25$ENSG00000157456 <- tmp$ENSG00000157456[match(cc_scores_z_long_25$uniqe_id, rownames(tmp))]
cc_scores_z_long_25$ENSG00000178999 <- tmp$ENSG00000178999[match(cc_scores_z_long_25$uniqe_id, rownames(tmp))]
```


```{r, fig.width=9, fig.height=6, echo = F}
plot_grid(
#MCM2, G1/S
ggplot(data=cc_scores_z_long_25, 
       aes(x=cluster, y=ENSG00000073111, col=cluster)) +
  geom_violin() +
  geom_boxplot(width=.1) +
  facet_wrap(~as.factor(chip_id), nrow=1) + ylab("log2CPM") + ggtitle("MCM2, G1/S"),
#MCM6, G1/S
ggplot(data=cc_scores_z_long_25, 
       aes(x=cluster, y=ENSG00000076003, col=cluster)) +
  geom_violin() +
  geom_boxplot(width=.1) +
  facet_wrap(~as.factor(chip_id), nrow=1) + ylab("log2CPM") +  ggtitle("MCM6, G1/S"),
#MCM5, G1/S
ggplot(data=cc_scores_z_long_25, 
       aes(x=cluster, y=ENSG00000100297, col=cluster)) +
  geom_violin() +
  geom_boxplot(width=.1) +
  facet_wrap(~as.factor(chip_id), nrow=1) + ylab("log2CPM") + ggtitle("MCM5, G1/S"),
nrow=3)

plot_grid(
#AURKA, M
ggplot(data=cc_scores_z_long_25, 
       aes(x=cluster, y=ENSG00000087586, col=cluster)) +
  geom_violin() +
  geom_boxplot(width=.1) +
  facet_wrap(~as.factor(chip_id), nrow=1) + ggtitle("AURKA, M"),
#CCNB2, M
ggplot(data=cc_scores_z_long_25, 
       aes(x=cluster, y=ENSG00000157456, col=cluster)) +
  geom_violin() +
  geom_boxplot(width=.1) +
  facet_wrap(~as.factor(chip_id), nrow=1) + ggtitle("CCNB2, M"),
#AURKB, G2
ggplot(data=cc_scores_z_long_25, 
       aes(x=cluster, y=ENSG00000178999, col=cluster)) +
  geom_violin() +
  geom_boxplot(width=.1) +
  facet_wrap(~as.factor(chip_id), nrow=1) + ggtitle("AURKB, G2"),
  nrow=3)
```

---

## Session information

```{r, echo = FALSE}
sessionInfo()
```



