---
title: "Normalize intensities across batches and positions"
author: "Joyce Hsiao"
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message=FALSE, warning=FALSE)
```

---

## Introduction/summary


In notations, 


$$
y_{ij} = \mu + \tau_i + \beta_j + \gamma_k + \epsilon_{ij}
$$
where $i = 1,2,..., I$ and $j = 1,2,..., J$. The parameters are estimated under sum-to-zero constraints $\sum \tau_i = 0$ and $\sum \beta_j = 0$.

Note that in this model 1) not all $y_{ij.}$ exists due to the incompleteness of the design, 2) the effects of individual and block are nonorthogonal, 3) the effects are additive due to the block design.



**TO DO: Apply batch correction prior to background correction??**


---

## Data and packages

$~$

```{r, echo = T}
library(data.table)
library(dplyr)
library(ggplot2)
library(cowplot)
library(RColorBrewer)
library(Biobase)
library(scales)
library(car)
library(ashr)
library(lsmeans)
```

Read in filtered data.

```{r}
df <- readRDS(file="../data/eset-filtered.rds")
pdata <- pData(df)
fdata <- fData(df)
```

---

## Source of variation

Statistical tests show that for GFP, there's significant individual effect, plate effect and position effect, and that for RFP and DAPI, there's no signficant individual effect or position effect but there's significant plate effect (all at P<.01).


```{r}
lm.rfp <- lm(rfp.median.log10sum~factor(chip_id)+factor(experiment) + factor(image_label), 
             data = pdata)
lm.gfp <- lm(gfp.median.log10sum~factor(chip_id)+factor(experiment) + factor(image_label), 
             data = pdata)
lm.dapi <- lm(dapi.median.log10sum~factor(chip_id)+factor(experiment) + factor(image_label), 
             data = pdata)

aov.lm.rfp <- Anova(lm.rfp, type = "III")
aov.lm.gfp <- Anova(lm.gfp, type = "III")
aov.lm.dapi <- Anova(lm.dapi, type = "III")
aov.lm.rfp
aov.lm.gfp
aov.lm.dapi
```

Indivdual and plate variation

```{r, fig.width=12, fig.height=8, echo = FALSE}
gfp.plot <- plot_grid(
  ggplot(pdata, 
          aes(x=chip_id, y=gfp.median.log10sum,
              col=chip_id)) +
    geom_violin() + geom_boxplot(width=.1),
  ggplot(pdata, 
        aes(x=factor(experiment), y=gfp.median.log10sum,
            col=factor(experiment))) +
  geom_violin() + geom_boxplot(width=.1), 
  ncol=2, labels = c("Individual variation", "Plate variation"))
title <- ggdraw() + draw_label("Green fluorescence intensites", fontface='bold')
plot_grid(title, gfp.plot, 
          ncol=1, rel_heights=c(0.1, 1)) # rel_heights values control title margins

## RFP
rfp.plot <- plot_grid(
  ggplot(pdata, 
          aes(x=chip_id, y=rfp.median.log10sum,
              col=chip_id)) +
    geom_violin() + geom_boxplot(width=.1),
  ggplot(pdata, 
        aes(x=factor(experiment), y=rfp.median.log10sum,
            col=factor(experiment))) +
  geom_violin() + geom_boxplot(width=.1), 
  ncol=2, labels = c("Individual variation", "Plate variation"))
title <- ggdraw() + draw_label("Red fluorescence intensites", fontface='bold')
plot_grid(title, rfp.plot, 
          ncol=1, rel_heights=c(0.1, 1)) # rel_heights values control title margins


## DAPI
dapi.plot <- plot_grid(
  ggplot(pdata, 
          aes(x=chip_id, y=dapi.median.log10sum,
              col=chip_id)) +
    geom_violin() + geom_boxplot(width=.1),
  ggplot(pdata, 
        aes(x=factor(experiment), y=dapi.median.log10sum,
            col=factor(experiment))) +
  geom_violin() + geom_boxplot(width=.1), 
  ncol=2, labels = c("Individual variation", "Plate variation"))
title <- ggdraw() + draw_label("DAPI intensites", fontface='bold')
plot_grid(title, dapi.plot, 
          ncol=1, rel_heights=c(0.1, 1)) # rel_heights values control title margins
```

Position variation

```{r}
well.gfp.median <- pdata %>% group_by(image_label) %>% summarize(., median(gfp.median.log10sum))
well.rfp.median <- pdata %>% group_by(image_label) %>% summarize(., median(rfp.median.log10sum))
well.dapi.median <- pdata %>% group_by(image_label) %>% summarize(., median(dapi.median.log10sum))

well.pp <- data.frame(well=pdata$well, image_label=pdata$image_label)
well.pp <- well.pp[!duplicated(well.pp),]
colbrew <- brewer.pal(9, "Set1")
well.pp$cols <- rep(colbrew[9], 96)
well.pp$cols[which(well.pp$well %in% c("A03", "A02", "A01", "A09", "A08", "A07"))] <- colbrew[1]
well.pp$cols[which(well.pp$well %in% c("H03", "H02", "H01", "H09", "H08", "H07"))] <- colbrew[2]
well.pp$cols[which(well.pp$well %in% c("A06", "A05", "A04", "A12", "A11", "A10"))] <- colbrew[3]
well.pp$cols[which(well.pp$well %in% c("H06", "H05", "H04", "H12", "H11", "H10"))] <- colbrew[4]
well.pp <- well.pp[order(well.pp$image_label),]

ord.gfp <- as.character(well.gfp.median$image_label[order(well.gfp.median$`median(gfp.median.log10sum)`)])
ord.rfp <- as.character(well.rfp.median$image_label[order(well.rfp.median$`median(rfp.median.log10sum)`)])
ord.dapi <- as.character(well.dapi.median$image_label[order(well.dapi.median$`median(dapi.median.log10sum)`)])
```

These are four corners previously found more likely to have high gene expression values in sequencing data.

```{r}
par(mfrow=c(1,1))
plot(1:7, 1:7, pch="", axes=F, ann=F)
legend("center", legend = c("A_a", "H_a", "A_b", "H_b"), col=colbrew[c(1,2,3,4)],
       pch=16)
```

```{r, fig.width=10, fig.height=8}
par(mfrow=c(3,1))
boxplot(rfp.median.log10sum ~ factor(image_label, levels=ord.rfp), 
        data=pdata, ylab = "RFP",
        col=well.pp$cols[as.numeric(ord.rfp)])
abline(h=0, lwd=2, col="royalblue")
boxplot(gfp.median.log10sum ~ factor(image_label, levels=ord.gfp), 
        data=pdata, ylab = "GFP",
        col=well.pp$cols[as.numeric(ord.gfp)])
abline(h=0, lwd=2, col="royalblue")
boxplot(dapi.median.log10sum ~ factor(image_label, levels=ord.dapi), 
        data=pdata, ylab = "GFP",
        col=well.pp$cols[as.numeric(ord.dapi)])
abline(h=0, lwd=2, col="royalblue")
title("Position variation", outer=TRUE, line = -1)
```


---

## Estimate effects

Contrast test to estimate effects for for plate and position ID.

```{r}
# make contrast matrix for plates
# each plate is compared to the average
n_plates <- uniqueN(pdata$experiment)
contrast_plates <- matrix(-1, nrow=n_plates, ncol=n_plates)
diag(contrast_plates) <- n_plates-1

# make contrast matrix for individuals
# each individual is compared to the average
n_pos <- uniqueN(pdata$image_label)
contrast_pos <- matrix(-1, nrow=n_pos, ncol=n_pos)
diag(contrast_pos) <- n_pos-1
```


```{r}
gfp.plates <- summary(lsmeans(lm.gfp, specs="experiment", contrast=contrast_plates))
gfp.pos <- summary(lsmeans(lm.gfp, specs="image_label", contrast=contrast_pos))

rfp.plates <- summary(lsmeans(lm.rfp, specs="experiment", contrast=contrast_plates))
rfp.pos <- summary(lsmeans(lm.rfp, specs="image_label", contrast=contrast_pos))

dapi.plates <- summary(lsmeans(lm.dapi, specs="experiment", contrast=contrast_plates))
dapi.pos <- summary(lsmeans(lm.dapi, specs="image_label", contrast=contrast_pos))
```


Substract plate effect from the raw estimates.

```{r compute-adjusted, eval = T}
## RFP
pdata$rfp.median.log10sum.adjust <- pdata$rfp.median.log10sum
rfp.plates$experiment <- as.character(rfp.plates$experiment)
rfp.pos$experiment <- as.character(rfp.pos$image_label)
pdata$experiment <- as.character(pdata$experiment)

exps <- unique(pdata$experiment)
for (i in 1:uniqueN(exps)) {
  exp <- exps[i]
  ii_exp <- which(pdata$experiment == exp)
  est_exp <- rfp.plates$lsmean[which(rfp.plates$experiment==exp)]
  pdata$rfp.median.log10sum.adjust[ii_exp] <- (pdata$rfp.median.log10sum[ii_exp] - est_exp)
}

pos <- unique(pdata$image_label)
for (i in 1:uniqueN(pos)) {
  p <- pos[i]
  ii_pos <- which(pdata$image_label == p)
  est_pos <- rfp.pos$lsmean[which(rfp.pos$image_label==p)]
  pdata$rfp.median.log10sum.adjust[ii_pos] <- (pdata$rfp.median.log10sum[ii_pos] - est_pos)
}


## GFP
pdata$gfp.median.log10sum.adjust <- pdata$gfp.median.log10sum
gfp.plates$experiment <- as.character(gfp.plates$experiment)
gfp.pos$experiment <- as.character(gfp.pos$image_label)
pdata$experiment <- as.character(pdata$experiment)

exps <- unique(pdata$experiment)
for (i in 1:uniqueN(exps)) {
  exp <- exps[i]
  ii_exp <- which(pdata$experiment == exp)
  est_exp <- gfp.plates$lsmean[which(gfp.plates$experiment==exp)]
  pdata$gfp.median.log10sum.adjust[ii_exp] <- (pdata$gfp.median.log10sum[ii_exp] - est_exp)
}

pos <- unique(pdata$image_label)
for (i in 1:uniqueN(pos)) {
  p <- pos[i]
  ii_pos <- which(pdata$image_label == p)
  est_pos <- gfp.pos$lsmean[which(gfp.pos$image_label==p)]
  pdata$gfp.median.log10sum.adjust[ii_pos] <- (pdata$gfp.median.log10sum[ii_pos] - est_pos)
}



## DAPI
pdata$dapi.median.log10sum.adjust <- pdata$dapi.median.log10sum
dapi.plates$experiment <- as.character(dapi.plates$experiment)
dapi.pos$experiment <- as.character(dapi.pos$image_label)
pdata$experiment <- as.character(pdata$experiment)

exps <- unique(pdata$experiment)
for (i in 1:uniqueN(exps)) {
  exp <- exps[i]
  ii_exp <- which(pdata$experiment == exp)
  est_exp <- dapi.plates$lsmean[which(dapi.plates$experiment==exp)]
  pdata$dapi.median.log10sum.adjust[ii_exp] <- (pdata$dapi.median.log10sum[ii_exp] - est_exp)
}

pos <- unique(pdata$image_label)
for (i in 1:uniqueN(pos)) {
  p <- pos[i]
  ii_pos <- which(pdata$image_label == p)
  est_pos <- dapi.pos$lsmean[which(dapi.pos$image_label==p)]
  pdata$dapi.median.log10sum.adjust[ii_pos] <- (pdata$dapi.median.log10sum[ii_pos] - est_pos)
}
```


After adjustment

```{r boxplot-adjusted, fig.width=12, fig.height=8, echo = FALSE}
gfp.plot <- plot_grid(
  ggplot(pdata, 
          aes(x=chip_id, y=gfp.median.log10sum.adjust,
              col=chip_id)) +
    geom_violin() + geom_boxplot(width=.1),
  ggplot(pdata, 
        aes(x=factor(experiment), y=gfp.median.log10sum.adjust,
            col=factor(experiment))) +
  geom_violin() + geom_boxplot(width=.1), 
  ncol=2, labels = c("Individual variation", "Plate variation"))
title <- ggdraw() + draw_label("Adjusted Green fluorescence intensites", fontface='bold')
plot_grid(title, gfp.plot, 
          ncol=1, rel_heights=c(0.1, 1)) # rel_heights values control title margins

## RFP
rfp.plot <- plot_grid(
  ggplot(pdata, 
          aes(x=chip_id, y=rfp.median.log10sum.adjust,
              col=chip_id)) +
    geom_violin() + geom_boxplot(width=.1),
  ggplot(pdata, 
        aes(x=factor(experiment), y=rfp.median.log10sum.adjust,
            col=factor(experiment))) +
  geom_violin() + geom_boxplot(width=.1), 
  ncol=2, labels = c("Individual variation", "Plate variation"))
title <- ggdraw() + draw_label("Adjusted Red fluorescence intensites", fontface='bold')
plot_grid(title, rfp.plot, 
          ncol=1, rel_heights=c(0.1, 1)) # rel_heights values control title margins


## DAPI
dapi.plot <- plot_grid(
  ggplot(pdata, 
          aes(x=chip_id, y=dapi.median.log10sum.adjust,
              col=chip_id)) +
    geom_violin() + geom_boxplot(width=.1),
  ggplot(pdata, 
        aes(x=factor(experiment), y=dapi.median.log10sum.adjust,
            col=factor(experiment))) +
  geom_violin() + geom_boxplot(width=.1), 
  ncol=2, labels = c("Individual variation", "Plate variation"))
title <- ggdraw() + draw_label("Adjusted DAPI intensites", fontface='bold')
plot_grid(title, dapi.plot, 
          ncol=1, rel_heights=c(0.1, 1)) # rel_heights values control title margins


## Position variation
well.gfp.median <- pdata %>% group_by(image_label) %>% summarize(., median(gfp.median.log10sum.adjust))
well.rfp.median <- pdata %>% group_by(image_label) %>% summarize(., median(rfp.median.log10sum.adjust))
well.dapi.median <- pdata %>% group_by(image_label) %>% summarize(., median(dapi.median.log10sum.adjust))

well.pp <- data.frame(well=pdata$well, image_label=pdata$image_label)
well.pp <- well.pp[!duplicated(well.pp),]
colbrew <- brewer.pal(9, "Set1")
well.pp$cols <- rep(colbrew[9], 96)
well.pp$cols[which(well.pp$well %in% c("A03", "A02", "A01", "A09", "A08", "A07"))] <- colbrew[1]
well.pp$cols[which(well.pp$well %in% c("H03", "H02", "H01", "H09", "H08", "H07"))] <- colbrew[2]
well.pp$cols[which(well.pp$well %in% c("A06", "A05", "A04", "A12", "A11", "A10"))] <- colbrew[3]
well.pp$cols[which(well.pp$well %in% c("H06", "H05", "H04", "H12", "H11", "H10"))] <- colbrew[4]
well.pp <- well.pp[order(well.pp$image_label),]

ord.gfp <- as.character(well.gfp.median$image_label[order(well.gfp.median$`median(gfp.median.log10sum.adjust)`)])
ord.rfp <- as.character(well.rfp.median$image_label[order(well.rfp.median$`median(rfp.median.log10sum.adjust)`)])
ord.dapi <- as.character(well.dapi.median$image_label[order(well.dapi.median$`median(dapi.median.log10sum.adjust)`)])
```


```{r}
## These are four corners previously found more likely to have high gene expression values in sequencing data.
par(mfrow=c(1,1))
plot(1:7, 1:7, pch="", axes=F, ann=F)
legend("center", legend = c("A_a", "H_a", "A_b", "H_b"), col=colbrew[c(1,2,3,4)],
       pch=16)
```


```{r plot-position-adjusted, fig.width=10, fig.height=9}
par(mfrow=c(3,1))
boxplot(rfp.median.log10sum.adjust ~ factor(image_label, levels=ord.rfp), 
        data=pdata, ylab = "RFP",
        col=well.pp$cols[as.numeric(ord.rfp)])
abline(h=0, lwd=2, col="royalblue")
boxplot(gfp.median.log10sum.adjust ~ factor(image_label, levels=ord.gfp), 
        data=pdata, ylab = "GFP",
        col=well.pp$cols[as.numeric(ord.gfp)])
abline(h=0, lwd=2, col="royalblue")
boxplot(dapi.median.log10sum.adjust ~ factor(image_label, levels=ord.dapi), 
        data=pdata, ylab = "GFP",
        col=well.pp$cols[as.numeric(ord.dapi)])
abline(h=0, lwd=2, col="royalblue")
title("Position variation", outer=TRUE, line = -1)
```


---

## Output results


Save corrected data to a temporary output folder.

```{r save-rds, eval = F, echo = T}
saveRDS(pdata, file = "../output/images-normalize-anova.Rmd/pdata.adj.rds")
```



---

## ash

apply ash to position estimates

```{r ash-position, fig.height=6, fig.width=7}
gfp.pos.ash <- ash(gfp.pos$lsmean, gfp.pos$SE, mixcompdist = "+uniform",
                   lik = lik_t(df=gfp.pos$df[1]))
rfp.pos.ash <- ash(rfp.pos$lsmean, rfp.pos$SE, mixcompdist = "+uniform",
                   lik = lik_t(df=rfp.pos$df[1]))
dapi.pos.ash <- ash(dapi.pos$lsmean, dapi.pos$SE, mixcompdist = "+uniform",
                    lik = lik_t(df=dapi.pos$df[1]))
  
par(mfrow=c(2,2))
plot(gfp.pos.ash$result$betahat, gfp.pos.ash$result$PosteriorMean,
     xlab = "beta hat", ylab = "Shrunken estimate", main = "GFP")
abline(0,1, col = "royalblue")
plot(rfp.pos.ash$result$betahat, rfp.pos.ash$result$PosteriorMean,
     xlab = "beta hat", ylab = "Shrunken estimate", main = "RFP")
abline(0,1, col = "royalblue")
plot(dapi.pos.ash$result$betahat, dapi.pos.ash$result$PosteriorMean,
     xlab = "beta hat", ylab = "Shrunken estimate", main = "DAPI")
abline(0,1, col = "royalblue")
     
par(mfrow=c(2,2))
plot(gfp.pos.ash$result$sebetahat, gfp.pos.ash$result$PosteriorSD,
     xlab = "Standard Error", ylab = "Shrunken estimate", main = "GFP")
abline(0,1, col = "royalblue")
plot(rfp.pos.ash$result$sebetahat, rfp.pos.ash$result$PosteriorSD,
     xlab = "Standard Error", ylab = "Shrunken estimate", main = "RFP")
abline(0,1, col = "royalblue")
plot(dapi.pos.ash$result$sebetahat, dapi.pos.ash$result$PosteriorSD,
     xlab = "Standard Error", ylab = "Shrunken estimate", main = "DAPI")
abline(0,1, col = "royalblue")
```

Plate effect. With only 16 plates, the effects can't be estimated...

```{r ash-plate, fig.height=6, fig.width=7, eval = F}
library(ashr)
gfp.plates.ash <- ash(gfp.plates$lsmean, gfp.plates$SE, mixcompdist = "+uniform",
                   lik = lik_t(df=gfp.plates$df[1]))
rfp.plates.ash <- ash(rfp.plates$lsmean, rfp.plates$SE, mixcompdist = "+uniform",
                   lik = lik_t(df=rfp.plates$df[1]))
dapi.plates.ash <- ash(dapi.plates$lsmean, dapi.plates$SE, mixcompdist = "+uniform",
                    lik = lik_t(df=dapi.plates$df[1]))
  
par(mfrow=c(2,2))
plot(gfp.plates.ash$result$betahat, gfp.plates.ash$result$PosteriorMean,
     xlab = "beta hat", ylab = "Shrunken estimate", main = "GFP")
abline(0,1, col = "royalblue")
plot(rfp.plates.ash$result$betahat, rfp.plates.ash$result$PosteriorMean,
     xlab = "beta hat", ylab = "Shrunken estimate", main = "RFP")
abline(0,1, col = "royalblue")
plot(dapi.plates.ash$result$betahat, dapi.plates.ash$result$PosteriorMean,
     xlab = "beta hat", ylab = "Shrunken estimate", main = "DAPI")
abline(0,1, col = "royalblue")
     
par(mfrow=c(2,2))
plot(gfp.plates.ash$result$sebetahat, gfp.plates.ash$result$PosteriorSD,
     xlab = "Standard Error", ylab = "Shrunken estimate", main = "GFP")
abline(0,1, col = "royalblue")
plot(rfp.plates.ash$result$sebetahat, rfp.plates.ash$result$PosteriorSD,
     xlab = "Standard Error", ylab = "Shrunken estimate", main = "RFP")
abline(0,1, col = "royalblue")
plot(dapi.plates.ash$result$sebetahat, dapi.plates.ash$result$PosteriorSD,
     xlab = "Standard Error", ylab = "Shrunken estimate", main = "DAPI")
abline(0,1, col = "royalblue")
```






---

## Session information

```{r, echo = FALSE}
sessionInfo()
```

