---
title: "Home"
output: 
  html_document:
    toc: TRUE
    toc_float: FALSE
---

---

## Overview

* [Data description](data-overview.html)

---

## Microscopy image analysis

* [Processing images - from images to intensities](images-process.html)

We evaluated and pre-processed the results of image analysis as follows:

1. We visually inspect images deteced to have none or more than one nucleus. For cases that are inconsistent with visual inspection, we correct the number of nuclei detected.  
    * [Inspect images with multiple nuclei](images-multiple-nuclei.html)
    * [Inspect images with no nucleus](images-zero-nuclei.html)
        
2. We applied background correction to the intensity measurements of GFP, RFP and DAPI based on the following analyses.  
    * [CONFESS results](confess-prelim.html)
    * [QC analysis including no. nuclei detected, DAPI, and intensity variation](images-qc.html)
    * [Explore using log10 sum pixel intensity for signal metrics](images-qc-followup.html)
    * [Compare correction approaches using median versus mean background](images-metrics.html)
    * [Explore associations between nucleus shape metrics vs intensities](images-metrics-cell-shape.html)

3. We analyzed intensity variation across individuals and batches and determined on an approach that removes batch effect in the data.  
    * [Visualize signal variation by plate and individual identity](images-qc-labels.html)
    * [Visualize the structure of signal variation by individual identity](images-qc-variation.html)
    * [Quantile normalization for GFP, RFP and DAPI](images-normalize-quantile.html)
    * [Estimate variance explained in IBD and correct for batch effects in intensities](images-normalize-anova.html)
    
---

## RNA-seq data preprcessing

1. The first step in preprocessing RNA-seq data consists of QC and filtering. 
    * Sample QC and filtering
        * [Sample QC criteria](sampleqc.html)  
        * [Sequencing depth](totals.html)  
        * [Reads versus molecules](reads-v-molecules.html)  
    
    * Gene QC and filtering
        * [gene filtering](gene-filtering.html)
        * [PCA with technical fators](pca-tf.html)

2. We then analyzed and corrected for batch effect due to C1 plate in the sequencing data
    * [Estimate variance explained in IBD and correct for batch effects](seqdata-batch-correction.html)


---

## Cell cycle signal in gene expression data

1. We investigated cell cycle signals in the sequencing data alone.
    
  * Consider [transgene count in sequencing data](images-transgene.html)
  * Compute linear correlation between gene expression levels [with DAPI and FUCCI intensities](images-seq-correlation.html)
  
2. We then assign categorical labels of cell cycle and explored the expresson profiles of these categories. 

  * Cluster samples by [Partition around medoids(PAM)](images-pam.html)
  * Cluster samples by [Guassian mixture modeling](images-mclust.html)
  * Select a subset of samples that are closet to the cluster centers (cluster representives) [using silhouette index](images-subset-silhouette.html)
  * Examine gene pression scores defined by the Macosko paper in the selected cluster representatives [before confounding correction](images-classify-fucci.html) and [after confounding correction](images-classify-fucci-adjusted.html)
  * Examine gene expression scores defined by the Macosk paer [in the sorted cells of the Leng et al. 2015 paper](images-classify-leng.html)

3. We ordered cells on a circle using FUCCI intensities alone. 

  * First, I used GFP and RFP intensities to estimate a [least-square fit of unit circle which approximate the relative ordering of cells on cell cycle](images-circle-ordering.html)
  * I computed the PCs of GFP and RFP and used these to infer the relative ordering on a circle (i.e., polar coordinates), and [to evaluate the circle fit, I computed circular-circular and circular-linear correlation with DAPI and gene expression](images-circle-ordering-eval.html)
  * Here I put together lists of genes identified as signfiicantly correlatd with the PC-based fit [by linear correlation and circular-linear correlation, also cyclical genes by smash](images-circle-ordering-sigcorgenes.html)



---

## Model fitting

* CellcycleR 0.1.6
    * [Model convergence assessment](images-cellcycleR-convergence.html)
    * [Fitting on intensities across plates and individuals](images-cellcycleR.html)
    * [Fitting on Leng data)](cellcycler-seqdata-leng.html)
    * [Fitting on fucci-seq RNA-seq data)](cellcycler-seqdata-fucci.html)
    
* Correlation for circular response variable: some simulations to check [its property and also relations with Fisher's z transformation](circ-correlation-simulation.html)
    
* Nonparametric circular regression
    * First, evaluate smash for smoothing gene expression data and compare [results of smash with NPCirc](images-circle-ordering-npreg.html)

 
---

## One-time investigations

* Why some gene symbols (genes) correspond to [multiple Ensembl IDs?](ensembl.html)

* I selected a set of cell cycle [genes that belong to GO term Cell Cycle and looked at the overlap with the detected genes in our data.](seqdata-select-cellcyclegenes.html)


